<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Copy number variation estimation from scRNA-seq | HKU Single-cell Workshop (Modules 4.2 &amp; 5)</title>
  <meta name="description" content="Cell trajectory inference; Cellular genetic analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Copy number variation estimation from scRNA-seq | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Copy number variation estimation from scRNA-seq | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  
  <meta name="twitter:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

<meta name="author" content="Yuanhua Lab: Mingze Gao, Rongting Huang, Xianjie Huang, Aaron Kwok, Zhuoxuan Li, Chen Qiao, Yuanhua Huang contact: yuanhua@hku.hk" />


<meta name="date" content="2021-07-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="SNV-analysis.html"/>
<link rel="next" href="preprocess.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SingleCell Workshop 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents-of-this-part"><i class="fa fa-check"></i><b>1.1</b> Contents of this part</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#other-modules-of-this-workshop"><i class="fa fa-check"></i><b>1.2</b> Other modules of this workshop</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="env-pre.html"><a href="env-pre.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="2.1" data-path="env-pre.html"><a href="env-pre.html#r-packages"><i class="fa fa-check"></i><b>2.1</b> R packages</a></li>
<li class="chapter" data-level="2.2" data-path="env-pre.html"><a href="env-pre.html#python_packages"><i class="fa fa-check"></i><b>2.2</b> Python packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="env-pre.html"><a href="env-pre.html#conda_install_windows"><i class="fa fa-check"></i><b>2.2.1</b> Installation on Windows</a></li>
<li class="chapter" data-level="2.2.2" data-path="env-pre.html"><a href="env-pre.html#conda_install_linux"><i class="fa fa-check"></i><b>2.2.2</b> Installation on Linux</a></li>
<li class="chapter" data-level="2.2.3" data-path="env-pre.html"><a href="env-pre.html#conda_install_mac"><i class="fa fa-check"></i><b>2.2.3</b> Installation on macOS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="RNA-velocity.html"><a href="RNA-velocity.html"><i class="fa fa-check"></i><b>3</b> Trajectory inference and RNA velocity</a>
<ul>
<li class="chapter" data-level="3.1" data-path="RNA-velocity.html"><a href="RNA-velocity.html#introduction-of-trajectory-inference"><i class="fa fa-check"></i><b>3.1</b> Introduction of trajectory inference</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="RNA-velocity.html"><a href="RNA-velocity.html#diffusion-map"><i class="fa fa-check"></i><b>3.1.1</b> Diffusion map</a></li>
<li class="chapter" data-level="3.1.2" data-path="RNA-velocity.html"><a href="RNA-velocity.html#monocle"><i class="fa fa-check"></i><b>3.1.2</b> Monocle</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="RNA-velocity.html"><a href="RNA-velocity.html#rna-velocity"><i class="fa fa-check"></i><b>3.2</b> RNA velocity</a></li>
<li class="chapter" data-level="3.3" data-path="RNA-velocity.html"><a href="RNA-velocity.html#rna-kinetics"><i class="fa fa-check"></i><b>3.3</b> RNA kinetics</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="RNA-velocity.html"><a href="RNA-velocity.html#unspliced-rna-indicates-transcriptional-speed"><i class="fa fa-check"></i><b>3.3.1</b> Unspliced RNA indicates transcriptional speed</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="RNA-velocity.html"><a href="RNA-velocity.html#use-scvelo-package-within-r"><i class="fa fa-check"></i><b>3.4</b> Use scVelo package within R</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="RNA-velocity.html"><a href="RNA-velocity.html#environment-and-preprocessing"><i class="fa fa-check"></i><b>3.4.1</b> Environment and preprocessing</a></li>
<li class="chapter" data-level="3.4.2" data-path="RNA-velocity.html"><a href="RNA-velocity.html#getting-started"><i class="fa fa-check"></i><b>3.4.2</b> Getting started</a></li>
<li class="chapter" data-level="3.4.3" data-path="RNA-velocity.html"><a href="RNA-velocity.html#data-preprocessing"><i class="fa fa-check"></i><b>3.4.3</b> Data Preprocessing</a></li>
<li class="chapter" data-level="3.4.4" data-path="RNA-velocity.html"><a href="RNA-velocity.html#pseudotime-based-on-diffusion-map"><i class="fa fa-check"></i><b>3.4.4</b> Pseudotime based on Diffusion Map</a></li>
<li class="chapter" data-level="3.4.5" data-path="RNA-velocity.html"><a href="RNA-velocity.html#compute-velocity-and-velocity-graph"><i class="fa fa-check"></i><b>3.4.5</b> Compute velocity and velocity graph</a></li>
<li class="chapter" data-level="3.4.6" data-path="RNA-velocity.html"><a href="RNA-velocity.html#diffusion-map-pseudotime-with-velocity"><i class="fa fa-check"></i><b>3.4.6</b> Diffusion-map Pseudotime with velocity</a></li>
<li class="chapter" data-level="3.4.7" data-path="RNA-velocity.html"><a href="RNA-velocity.html#plot-results"><i class="fa fa-check"></i><b>3.4.7</b> Plot Results</a></li>
<li class="chapter" data-level="3.4.8" data-path="RNA-velocity.html"><a href="RNA-velocity.html#interprete-velocity"><i class="fa fa-check"></i><b>3.4.8</b> Interprete Velocity</a></li>
<li class="chapter" data-level="3.4.9" data-path="RNA-velocity.html"><a href="RNA-velocity.html#velocity-in-cycling-progenitors"><i class="fa fa-check"></i><b>3.4.9</b> Velocity in cycling progenitors</a></li>
<li class="chapter" data-level="3.4.10" data-path="RNA-velocity.html"><a href="RNA-velocity.html#dynamical-mode-and-related-analysis"><i class="fa fa-check"></i><b>3.4.10</b> Dynamical Mode and related analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="SNV-analysis.html"><a href="SNV-analysis.html"><i class="fa fa-check"></i><b>4</b> Somatic mutation analysis in single cells</a>
<ul>
<li class="chapter" data-level="4.1" data-path="SNV-analysis.html"><a href="SNV-analysis.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="SNV-analysis.html"><a href="SNV-analysis.html#choice-of-protocols"><i class="fa fa-check"></i><b>4.1.1</b> Choice of protocols</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="SNV-analysis.html"><a href="SNV-analysis.html#snv-analysis"><i class="fa fa-check"></i><b>4.2</b> SNV analysis</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="SNV-analysis.html"><a href="SNV-analysis.html#call-somatic-variants-and-genotype-single-cells"><i class="fa fa-check"></i><b>4.2.1</b> Call somatic variants and genotype single cells</a></li>
<li class="chapter" data-level="4.2.2" data-path="SNV-analysis.html"><a href="SNV-analysis.html#visualize-variat-allele-fequency"><i class="fa fa-check"></i><b>4.2.2</b> Visualize variat allele fequency</a></li>
<li class="chapter" data-level="4.2.3" data-path="SNV-analysis.html"><a href="SNV-analysis.html#use-extra-information-on-clonal-tree"><i class="fa fa-check"></i><b>4.2.3</b> Use extra information on clonal tree</a></li>
<li class="chapter" data-level="4.2.4" data-path="SNV-analysis.html"><a href="SNV-analysis.html#run-cardelino-for-inferring-clonal-structure"><i class="fa fa-check"></i><b>4.2.4</b> Run cardelino for inferring clonal structure</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="SNV-analysis.html"><a href="SNV-analysis.html#mtsnv-analysis"><i class="fa fa-check"></i><b>4.3</b> mtSNV analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="SNV-analysis.html"><a href="SNV-analysis.html#visualise-allele-frequency"><i class="fa fa-check"></i><b>4.3.1</b> Visualise allele frequency</a></li>
<li class="chapter" data-level="4.3.2" data-path="SNV-analysis.html"><a href="SNV-analysis.html#infer-clonal-structure-with-mtdna-variants"><i class="fa fa-check"></i><b>4.3.2</b> Infer clonal structure with mtDNA variants</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="CNV-analysis.html"><a href="CNV-analysis.html"><i class="fa fa-check"></i><b>5</b> Copy number variation estimation from scRNA-seq</a>
<ul>
<li class="chapter" data-level="5.1" data-path="CNV-analysis.html"><a href="CNV-analysis.html#method-infercnv"><i class="fa fa-check"></i><b>5.1</b> Method: inferCNV</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="CNV-analysis.html"><a href="CNV-analysis.html#install-infercnv"><i class="fa fa-check"></i><b>5.1.1</b> install inferCNV</a></li>
<li class="chapter" data-level="5.1.2" data-path="CNV-analysis.html"><a href="CNV-analysis.html#getting-started-1"><i class="fa fa-check"></i><b>5.1.2</b> getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="CNV-analysis.html"><a href="CNV-analysis.html#application-on-tnbc1"><i class="fa fa-check"></i><b>5.2</b> Application on TNBC1</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="CNV-analysis.html"><a href="CNV-analysis.html#data-description"><i class="fa fa-check"></i><b>5.2.1</b> data description</a></li>
<li class="chapter" data-level="5.2.2" data-path="CNV-analysis.html"><a href="CNV-analysis.html#run-infercnv"><i class="fa fa-check"></i><b>5.2.2</b> run inferCNV</a></li>
<li class="chapter" data-level="5.2.3" data-path="CNV-analysis.html"><a href="CNV-analysis.html#infercnv-result"><i class="fa fa-check"></i><b>5.2.3</b> inferCNV result</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="CNV-analysis.html"><a href="CNV-analysis.html#ref"><i class="fa fa-check"></i><b>5.3</b> ref</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="preprocess.html"><a href="preprocess.html"><i class="fa fa-check"></i><b>6</b> Preprocessing of dataset</a>
<ul>
<li class="chapter" data-level="6.1" data-path="preprocess.html"><a href="preprocess.html#preprocessing-for-rna-velocity"><i class="fa fa-check"></i><b>6.1</b> Preprocessing for RNA Velocity</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="preprocess.html"><a href="preprocess.html#list-of-packages-aligned-in-pipeline"><i class="fa fa-check"></i><b>6.1.1</b> List of packages aligned in pipeline:</a></li>
<li class="chapter" data-level="6.1.2" data-path="preprocess.html"><a href="preprocess.html#installation-of-packagessoftwares"><i class="fa fa-check"></i><b>6.1.2</b> Installation of packages/softwares</a></li>
<li class="chapter" data-level="6.1.3" data-path="preprocess.html"><a href="preprocess.html#preprocessing-pipline"><i class="fa fa-check"></i><b>6.1.3</b> Preprocessing pipline</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="preprocess.html"><a href="preprocess.html#preprocessing-for-somatic-mutation-analysis"><i class="fa fa-check"></i><b>6.2</b> Preprocessing for somatic mutation analysis</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="preprocess.html"><a href="preprocess.html#pileup-with-cellsnp-lite"><i class="fa fa-check"></i><b>6.2.1</b> Pileup with cellsnp-lite</a></li>
<li class="chapter" data-level="6.2.2" data-path="preprocess.html"><a href="preprocess.html#clonal-analysis-with-mquad"><i class="fa fa-check"></i><b>6.2.2</b> Clonal analysis with MQuad</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="preprocess.html"><a href="preprocess.html#wsl_install"><i class="fa fa-check"></i><b>6.3</b> (Optional) Install Windows Subsystem for Linux</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="preprocess.html"><a href="preprocess.html#what-is-the-windows-subsystem-for-linux-wsl"><i class="fa fa-check"></i><b>6.3.1</b> What is the Windows Subsystem for Linux (WSL)?</a></li>
<li class="chapter" data-level="6.3.2" data-path="preprocess.html"><a href="preprocess.html#manual-installation-steps"><i class="fa fa-check"></i><b>6.3.2</b> Manual Installation Steps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">HKU Single-cell Workshop (Modules 4.2 &amp; 5)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="CNV_analysis" class="section level1" number="5">
<h1><span class="header-section-number">Chapter 5</span> Copy number variation estimation from scRNA-seq</h1>
<p>Rongting Huang</p>
<p>2021-06-29</p>
<div id="method-infercnv" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Method: inferCNV</h2>
<p><a href="https://github.com/broadinstitute/inferCNV/wiki">InferCNV: Inferring copy number alterations from tumor single cell RNA-Seq data</a></p>
<div id="install-infercnv" class="section level3" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> install inferCNV</h3>
<p><strong>Software Requirements</strong></p>
<ul>
<li><a href="https://mcmc-jags.sourceforge.io/">JAGS</a></li>
<li>R (&gt;3.6)</li>
</ul>
<p>In order to run infercnv, JAGS (Just Another Gibbs Sampler) must be installed.</p>
<p>Download JAGS from <a href="https://sourceforge.net/projects/mcmc-jags/files/JAGS/4.x/" class="uri">https://sourceforge.net/projects/mcmc-jags/files/JAGS/4.x/</a> and install JAGS in your environment (windows/MAC).</p>
<p>If you use inferCNV on server, install JAGS via <code>conda install</code> in your conda environment is recommended.</p>
<pre><code>conda install -c conda-forge jags</code></pre>
<p>More details refer to <a href="https://github.com/broadinstitute/inferCNV/wiki">inferCNV wiki page</a></p>
<p><strong>Five options for installing inferCNV</strong></p>
<p>Option A: Install infercnv from BioConductor (preferred)</p>
<pre><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
     install.packages(&quot;BiocManager&quot;)
BiocManager::install(&quot;infercnv&quot;)</code></pre>
<p>For more other options, refer to <a href="https://github.com/broadinstitute/inferCNV/wiki/Installing-infercnv">Five options for installing inferCNV</a></p>
<p><strong>Data requirements</strong></p>
<ul>
<li>a raw counts matrix of single-cell RNA-Seq expression</li>
<li>an annotations file which indicates which cells are tumor vs. normal.</li>
<li>a gene/chromosome positions file</li>
</ul>
<p><a href="https://github.com/broadinstitute/inferCNV/wiki/File-Definitions">File-Definitions</a></p>
</div>
<div id="getting-started-1" class="section level3" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> getting started</h3>
<p>If you have installed infercnv from BioConductor, you can run the example data with:</p>
<pre><code>library(infercnv)

infercnv_obj = CreateInfercnvObject(raw_counts_matrix=system.file(&quot;extdata&quot;, &quot;oligodendroglioma_expression_downsampled.counts.matrix.gz&quot;, package = &quot;infercnv&quot;),
                                    annotations_file=system.file(&quot;extdata&quot;, &quot;oligodendroglioma_annotations_downsampled.txt&quot;, package = &quot;infercnv&quot;),
                                    delim=&quot;\t&quot;,
                                    gene_order_file=system.file(&quot;extdata&quot;, &quot;gencode_downsampled.EXAMPLE_ONLY_DONT_REUSE.txt&quot;, package = &quot;infercnv&quot;),
                                    ref_group_names=c(&quot;Microglia/Macrophage&quot;,&quot;Oligodendrocytes (non-malignant)&quot;)) 

infercnv_obj = infercnv::run(infercnv_obj,
                             cutoff=1, # cutoff=1 works well for Smart-seq2, and cutoff=0.1 works well for 10x Genomics
                             out_dir=tempfile(), 
                             cluster_by_groups=TRUE, 
                             denoise=TRUE,
                             HMM=TRUE)</code></pre>
<p>If you can run the getting started part with demo data provided by inferCNV, then it is installed successfully.</p>
<p><strong>Demo Example Figure</strong></p>
<div class="figure">
<img src="CNV_analysis_file/figs/demo-infercnv.png" alt="" />
<p class="caption">Demo Example Figure</p>
</div>
</div>
</div>
<div id="application-on-tnbc1" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Application on TNBC1</h2>
<div id="data-description" class="section level3" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> data description</h3>
<p>TNBC1 is a triple negative breast cancer tumor sample of high tumor purity (72.6%) with 796 single tumor cells and 301 normal cells. The dataset is available on NCBI GEO under the accession number GSM4476486.</p>
<table>
<caption>Details of TNBC1 dataset (from published articles, copyKAT).</caption>
<colgroup>
<col width="28%" />
<col width="18%" />
<col width="21%" />
<col width="32%" />
</colgroup>
<thead>
<tr class="header">
<th>TNBC1</th>
<th>Number of clones</th>
<th>Number of tumor clones</th>
<th>Tumor clone-specific copy gain</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Triple negative breast cancer</td>
<td><pre><code>    3</code></pre></td>
<td><pre><code>        2</code></pre></td>
<td><ul>
<li>C1: 4p, 7q, 9, 17q</li>
<li>C2: 3p, 6q, 7p, 11q, X</li>
</ul></td>
</tr>
</tbody>
</table>
<ul>
<li>Expression</li>
</ul>
<table>
<caption>Subclusters of TNBC1 dataset (from gene expression analysis-Seurat).</caption>
<thead>
<tr class="header">
<th>Clone A</th>
<th>Clone B</th>
<th>Normal</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>488</td>
<td>307</td>
<td>302</td>
</tr>
</tbody>
</table>
<p>Notes: 488,307, 247, 55</p>
<div class="figure">
<img src="CNV_analysis_file/figs/tnbc1-exp/tnbc1-umap-3clusters.jpeg" alt="" />
<p class="caption">umap</p>
</div>
<ul>
<li>B Allele Frenquency (BAF)</li>
</ul>
<div class="figure">
<embed src="./CNV_analysis_file/figs/tnbc1-baf/0sub_chr.pdf" style="width:120.0%" />
<p class="caption">baf1</p>
</div>
<div class="figure">
<embed src="./CNV_analysis_file/figs/tnbc1-baf/1sub_chr.pdf" style="width:120.0%" />
<p class="caption">baf2</p>
</div>
<div class="figure">
<embed src="./CNV_analysis_file/figs/tnbc1-baf/2sub_chr.pdf" style="width:120.0%" />
<p class="caption">baf3</p>
</div>
<ul>
<li>BAF V.S. Expression</li>
</ul>
<div class="figure">
<img src="CNV_analysis_file/figs/BAFvsExpression.png" alt="" />
<p class="caption">confusion heatmap</p>
</div>
</div>
<div id="run-infercnv" class="section level3" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> run inferCNV</h3>
<p><a href="https://sourceforge.net/projects/sgcellworkshop/files/CNV_analysis/CNV_analysis_data.zip/download">data_download</a></p>
<p><a href="https://github.com/Rongtingting/SingleCell-Workshop-2021/blob/master/CNV_analysis_file/demo1_inferCNV_TNBC1.log">demo1_log_file</a></p>
<p><a href="https://github.com/Rongtingting/SingleCell-Workshop-2021/blob/master/CNV_analysis_file/demo2_inferCNV_TNBC1.log">demo2_log_file</a></p>
<p><a href="https://github.com/Rongtingting/SingleCell-Workshop-2021/blob/master/CNV_analysis_file/demo2_outputfiles.lst">output_files</a></p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="CNV-analysis.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># library(infercnv)</span></span>
<span id="cb47-2"><a href="CNV-analysis.html#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="co"># library(utils)</span></span>
<span id="cb47-3"><a href="CNV-analysis.html#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="co"># library (BiocGenerics)</span></span>
<span id="cb47-4"><a href="CNV-analysis.html#cb47-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-5"><a href="CNV-analysis.html#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="do">## DEMO1</span></span>
<span id="cb47-6"><a href="CNV-analysis.html#cb47-6" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb47-7"><a href="CNV-analysis.html#cb47-7" aria-hidden="true" tabindex="-1"></a><span class="co"># tnbc &lt;- read.delim(&quot;C://Users/Rongting/Documents/GitHub_repos/combinedTNBC1.txt&quot;)</span></span>
<span id="cb47-8"><a href="CNV-analysis.html#cb47-8" aria-hidden="true" tabindex="-1"></a><span class="co"># anno &lt;- tnbc[2,]</span></span>
<span id="cb47-9"><a href="CNV-analysis.html#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="co"># anno &lt;- t(anno)</span></span>
<span id="cb47-10"><a href="CNV-analysis.html#cb47-10" aria-hidden="true" tabindex="-1"></a><span class="co"># anno &lt;- as.data.frame(anno)</span></span>
<span id="cb47-11"><a href="CNV-analysis.html#cb47-11" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb47-12"><a href="CNV-analysis.html#cb47-12" aria-hidden="true" tabindex="-1"></a><span class="co"># gex &lt;- tnbc[-c(1:2),]</span></span>
<span id="cb47-13"><a href="CNV-analysis.html#cb47-13" aria-hidden="true" tabindex="-1"></a><span class="co"># gex &lt;- type.convert(gex)</span></span>
<span id="cb47-14"><a href="CNV-analysis.html#cb47-14" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb47-15"><a href="CNV-analysis.html#cb47-15" aria-hidden="true" tabindex="-1"></a><span class="co"># gene_file &lt;- &quot;C://Users/Rongting/Documents/GitHub_repos/gene_note_noheader_unique.txt&quot;</span></span>
<span id="cb47-16"><a href="CNV-analysis.html#cb47-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-17"><a href="CNV-analysis.html#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb47-18"><a href="CNV-analysis.html#cb47-18" aria-hidden="true" tabindex="-1"></a><span class="co"># infercnv_obj = CreateInfercnvObject(raw_counts_matrix=gex,</span></span>
<span id="cb47-19"><a href="CNV-analysis.html#cb47-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                                     annotations_file=anno,</span></span>
<span id="cb47-20"><a href="CNV-analysis.html#cb47-20" aria-hidden="true" tabindex="-1"></a><span class="co">#                                     delim=&#39;\t&#39;,</span></span>
<span id="cb47-21"><a href="CNV-analysis.html#cb47-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                                     gene_order_file=gene_file,</span></span>
<span id="cb47-22"><a href="CNV-analysis.html#cb47-22" aria-hidden="true" tabindex="-1"></a><span class="co">#                                     ref_group_names= &quot;N&quot;)</span></span>
<span id="cb47-23"><a href="CNV-analysis.html#cb47-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-24"><a href="CNV-analysis.html#cb47-24" aria-hidden="true" tabindex="-1"></a><span class="co"># output = &quot;C://Users/Rongting/Documents/GitHub_repos/tnbc1_demo&quot;</span></span>
<span id="cb47-25"><a href="CNV-analysis.html#cb47-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-26"><a href="CNV-analysis.html#cb47-26" aria-hidden="true" tabindex="-1"></a><span class="co"># infercnv_obj = infercnv::run(infercnv_obj,</span></span>
<span id="cb47-27"><a href="CNV-analysis.html#cb47-27" aria-hidden="true" tabindex="-1"></a><span class="co">#                              cutoff=0.1,  </span></span>
<span id="cb47-28"><a href="CNV-analysis.html#cb47-28" aria-hidden="true" tabindex="-1"></a><span class="co">#                              out_dir= output , </span></span>
<span id="cb47-29"><a href="CNV-analysis.html#cb47-29" aria-hidden="true" tabindex="-1"></a><span class="co">#                              cluster_by_groups=T,   </span></span>
<span id="cb47-30"><a href="CNV-analysis.html#cb47-30" aria-hidden="true" tabindex="-1"></a><span class="co">#                              denoise=T,</span></span>
<span id="cb47-31"><a href="CNV-analysis.html#cb47-31" aria-hidden="true" tabindex="-1"></a><span class="co">#                              HMM=T)</span></span>
<span id="cb47-32"><a href="CNV-analysis.html#cb47-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-33"><a href="CNV-analysis.html#cb47-33" aria-hidden="true" tabindex="-1"></a><span class="do">## DEMO2</span></span>
<span id="cb47-34"><a href="CNV-analysis.html#cb47-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-35"><a href="CNV-analysis.html#cb47-35" aria-hidden="true" tabindex="-1"></a><span class="co"># gene_file &lt;- &quot;C://Users/Rongting/Documents/GitHub_repos/gene_note_noheader_unique.txt&quot;</span></span>
<span id="cb47-36"><a href="CNV-analysis.html#cb47-36" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb47-37"><a href="CNV-analysis.html#cb47-37" aria-hidden="true" tabindex="-1"></a><span class="co"># anno_file &lt;- &#39;C://Users/Rongting/Documents/GitHub_repos/tnbc-3cluster-id.txt&#39;</span></span>
<span id="cb47-38"><a href="CNV-analysis.html#cb47-38" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb47-39"><a href="CNV-analysis.html#cb47-39" aria-hidden="true" tabindex="-1"></a><span class="co"># infercnv_obj2 = CreateInfercnvObject(raw_counts_matrix=gex,</span></span>
<span id="cb47-40"><a href="CNV-analysis.html#cb47-40" aria-hidden="true" tabindex="-1"></a><span class="co">#                                     annotations_file=anno_file,</span></span>
<span id="cb47-41"><a href="CNV-analysis.html#cb47-41" aria-hidden="true" tabindex="-1"></a><span class="co">#                                     delim=&#39;\t&#39;,</span></span>
<span id="cb47-42"><a href="CNV-analysis.html#cb47-42" aria-hidden="true" tabindex="-1"></a><span class="co">#                                     gene_order_file=gene_file,</span></span>
<span id="cb47-43"><a href="CNV-analysis.html#cb47-43" aria-hidden="true" tabindex="-1"></a><span class="co">#                                     ref_group_names= &quot;Normal&quot;)</span></span>
<span id="cb47-44"><a href="CNV-analysis.html#cb47-44" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb47-45"><a href="CNV-analysis.html#cb47-45" aria-hidden="true" tabindex="-1"></a><span class="co"># output = &quot;C://Users/Rongting/Documents/GitHub_repos/tnbc1_demo2&quot;</span></span>
<span id="cb47-46"><a href="CNV-analysis.html#cb47-46" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb47-47"><a href="CNV-analysis.html#cb47-47" aria-hidden="true" tabindex="-1"></a><span class="co"># infercnv_obj2 = infercnv::run(infercnv_obj2,</span></span>
<span id="cb47-48"><a href="CNV-analysis.html#cb47-48" aria-hidden="true" tabindex="-1"></a><span class="co">#                              cutoff=0.1,</span></span>
<span id="cb47-49"><a href="CNV-analysis.html#cb47-49" aria-hidden="true" tabindex="-1"></a><span class="co">#                              out_dir= output,</span></span>
<span id="cb47-50"><a href="CNV-analysis.html#cb47-50" aria-hidden="true" tabindex="-1"></a><span class="co">#                              cluster_by_groups=T,</span></span>
<span id="cb47-51"><a href="CNV-analysis.html#cb47-51" aria-hidden="true" tabindex="-1"></a><span class="co">#                              denoise=T,</span></span>
<span id="cb47-52"><a href="CNV-analysis.html#cb47-52" aria-hidden="true" tabindex="-1"></a><span class="co">#                              HMM=T)</span></span></code></pre></div>
<pre><code>#################
##Notes
#################

## load the package
library(Seurat)
library(infercnv)

## prepare the data (cellranger output)

### load count matrix (example)
matrix_path &lt;- &quot;../cellranger/xxxx/count_xxxxx/outs/filtered_gene_bc_matrices/GRCh38/&quot;


### read count matrix
gex_mtx &lt;- Seurat::Read10X(data.dir = matrix_path)

### run inferCNV with loop
celltype = c(&#39;CloneA&#39;, &#39;CloneB&#39;, &#39;Normal&#39;)

for (i in celltype){
  infercnv_obj1 = CreateInfercnvObject(raw_counts_matrix=gex_mtx,
                                       annotations_file=anno_file,
                                       delim=&#39;\t&#39;,
                                       gene_order_file=gene_file,
                                       ref_group_names=c(i))
  
  output &lt;- paste0(&#39;/groups/cgsd/rthuang/processed_data/inferCNV/xxxx/&#39;,&#39;xxxx_&#39;, i)
  
  infercnv_obj1 = infercnv::run(infercnv_obj1,
                                cutoff=0.1,  
                                out_dir= output , 
                                cluster_by_groups=T,   
                                denoise=T,
                                HMM=T)
}
</code></pre>
</div>
<div id="infercnv-result" class="section level3" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> inferCNV result</h3>
<ul>
<li>demo1</li>
</ul>
<div class="figure">
<img src="CNV_analysis_file/figs/tnbc1_demo1/infercnv.png" alt="" />
<p class="caption">infercnv1</p>
</div>
<div class="figure">
<img src="CNV_analysis_file/figs/tnbc1_demo1/infercnv.18_HMM_pred.Bayes_Net.Pnorm_0.5.png" alt="" />
<p class="caption">infercnv2</p>
</div>
<ul>
<li>demo2</li>
</ul>
<div class="figure">
<img src="CNV_analysis_file/figs/tnbc1_demo2/infercnv.png" alt="" />
<p class="caption">infercnv3</p>
</div>
<div class="figure">
<img src="CNV_analysis_file/figs/tnbc1_demo2/infercnv.18_HMM_pred.Bayes_Net.Pnorm_0.5.png" alt="" />
<p class="caption">infercnv4</p>
</div>
</div>
</div>
<div id="ref" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> ref</h2>
<p><a href="https://www.r-bloggers.com/2012/04/getting-started-with-jags-rjags-and-bayesian-modelling/" class="uri">https://www.r-bloggers.com/2012/04/getting-started-with-jags-rjags-and-bayesian-modelling/</a></p>
<pre><code>bookdown::render_book(&quot;index.Rmd&quot;, &quot;bookdown::gitbook&quot;)</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="SNV-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="preprocess.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["SingleCell Workshop.pdf", "SingleCell Workshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
