<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Prerequisites | HKU Single-cell Workshop (Modules 4.2 &amp; 5)</title>
  <meta name="description" content="Cell trajectory inference; Cellular genetic analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Prerequisites | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Prerequisites | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  
  <meta name="twitter:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

<meta name="author" content="Yuanhua Lab: Mingze Gao, Rongting Huang, Xianjie Huang, Aaron Kwok, Zhuoxuan Li, Chen Qiao, Yuanhua Huang" />


<meta name="date" content="2021-07-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="pseudo-time-and-trajectory-analysis.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SingleCell Workshop 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents-of-this-part"><i class="fa fa-check"></i><b>1.1</b> Contents of this part</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#other-modules-of-this-workshop"><i class="fa fa-check"></i><b>1.2</b> Other modules of this workshop</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="env-pre.html"><a href="env-pre.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="2.1" data-path="env-pre.html"><a href="env-pre.html#quick-summary"><i class="fa fa-check"></i><b>2.1</b> Quick summary</a></li>
<li class="chapter" data-level="2.2" data-path="env-pre.html"><a href="env-pre.html#wsl_install"><i class="fa fa-check"></i><b>2.2</b> (Optional) Install Windows Subsystem for Linux</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="env-pre.html"><a href="env-pre.html#what-is-the-windows-subsystem-for-linux-wsl"><i class="fa fa-check"></i><b>2.2.1</b> What is the Windows Subsystem for Linux (WSL)?</a></li>
<li class="chapter" data-level="2.2.2" data-path="env-pre.html"><a href="env-pre.html#manual-installation-steps"><i class="fa fa-check"></i><b>2.2.2</b> Manual Installation Steps</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="env-pre.html"><a href="env-pre.html#conda_install"><i class="fa fa-check"></i><b>2.3</b> Install Conda Environment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="env-pre.html"><a href="env-pre.html#what-is-conda"><i class="fa fa-check"></i><b>2.3.1</b> What is conda?</a></li>
<li class="chapter" data-level="2.3.2" data-path="env-pre.html"><a href="env-pre.html#conda_install_windows"><i class="fa fa-check"></i><b>2.3.2</b> Installation on Windows</a></li>
<li class="chapter" data-level="2.3.3" data-path="env-pre.html"><a href="env-pre.html#conda_install_linux"><i class="fa fa-check"></i><b>2.3.3</b> Installation on Linux</a></li>
<li class="chapter" data-level="2.3.4" data-path="env-pre.html"><a href="env-pre.html#conda_install_mac"><i class="fa fa-check"></i><b>2.3.4</b> Installation on macOS</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="env-pre.html"><a href="env-pre.html#other-preparation"><i class="fa fa-check"></i><b>2.4</b> Other Preparation</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="env-pre.html"><a href="env-pre.html#conda-configuration"><i class="fa fa-check"></i><b>2.4.1</b> Conda Configuration</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="pseudo-time-and-trajectory-analysis.html"><a href="pseudo-time-and-trajectory-analysis.html"><i class="fa fa-check"></i><b>3</b> Pseudo-time and trajectory analysis</a></li>
<li class="chapter" data-level="4" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html"><i class="fa fa-check"></i><b>4</b> Trajectory Inference using RNA Velocity</a>
<ul>
<li class="chapter" data-level="4.1" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#use-scvelo-package-within-r"><i class="fa fa-check"></i><b>4.1</b> Use scVelo package within R</a></li>
<li class="chapter" data-level="4.2" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#endocrine-pancreas"><i class="fa fa-check"></i><b>4.2</b> Endocrine Pancreas</a></li>
<li class="chapter" data-level="4.3" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#data-preprocessing"><i class="fa fa-check"></i><b>4.3</b> Data Preprocessing</a></li>
<li class="chapter" data-level="4.4" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#pseudotime-based-on-diffusion-map"><i class="fa fa-check"></i><b>4.4</b> Pseudotime based on Diffusion Map</a></li>
<li class="chapter" data-level="4.5" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#compute-velocity-and-velocity-graph"><i class="fa fa-check"></i><b>4.5</b> Compute velocity and velocity graph</a></li>
<li class="chapter" data-level="4.6" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#diffusion-map-pseudotime-with-velocity"><i class="fa fa-check"></i><b>4.6</b> Diffusion-map Pseudotime with velocity</a></li>
<li class="chapter" data-level="4.7" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#plot-results"><i class="fa fa-check"></i><b>4.7</b> Plot Results</a></li>
<li class="chapter" data-level="4.8" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#interprete-velocity"><i class="fa fa-check"></i><b>4.8</b> Interprete Velocity</a></li>
<li class="chapter" data-level="4.9" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#velocity-in-cycling-progenitors"><i class="fa fa-check"></i><b>4.9</b> Velocity in cycling progenitors</a></li>
<li class="chapter" data-level="4.10" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#dynamical-mode-and-related-analysis"><i class="fa fa-check"></i><b>4.10</b> Dynamical Mode and related analysis</a></li>
<li class="chapter" data-level="4.11" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#session-info"><i class="fa fa-check"></i><b>4.11</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html"><i class="fa fa-check"></i><b>5</b> Copy number variation estimation from scRNA-seq</a>
<ul>
<li class="chapter" data-level="5.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#method-infercnv"><i class="fa fa-check"></i><b>5.1</b> Method: inferCNV</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#install-infercnv"><i class="fa fa-check"></i><b>5.1.1</b> install inferCNV</a></li>
<li class="chapter" data-level="5.1.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#getting-started"><i class="fa fa-check"></i><b>5.1.2</b> getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#application-on-tnbc1"><i class="fa fa-check"></i><b>5.2</b> Application on TNBC1</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#data-description"><i class="fa fa-check"></i><b>5.2.1</b> data description</a></li>
<li class="chapter" data-level="5.2.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#run-infercnv"><i class="fa fa-check"></i><b>5.2.2</b> run inferCNV</a></li>
<li class="chapter" data-level="5.2.3" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#infercnv-result"><i class="fa fa-check"></i><b>5.2.3</b> inferCNV result</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#ref"><i class="fa fa-check"></i><b>5.3</b> ref</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><i class="fa fa-check"></i><b>6</b> mtDNA variants for lineage inference from single-cell omics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html#pileup-with-cellsnp-lite"><i class="fa fa-check"></i><b>6.1</b> Pileup with cellsnp-lite</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html#pileup-without-given-snps"><i class="fa fa-check"></i><b>6.1.1</b> Pileup without given SNPs</a></li>
<li class="chapter" data-level="6.1.2" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html#pileup-with-given-snps"><i class="fa fa-check"></i><b>6.1.2</b> Pileup with given SNPs</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html#clonal-analysis-with-mquad"><i class="fa fa-check"></i><b>6.2</b> Clonal analysis with MQuad</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html"><i class="fa fa-check"></i><b>7</b> Preprocessing of dataset</a>
<ul>
<li class="chapter" data-level="7.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#list-of-packages-aligned-in-pipeline"><i class="fa fa-check"></i><b>7.1</b> List of packages aligned in pipeline:</a></li>
<li class="chapter" data-level="7.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#installation-of-packagessoftwares"><i class="fa fa-check"></i><b>7.2</b> Installation of packages/softwares</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#sratool"><i class="fa fa-check"></i><b>7.2.1</b> SRATool</a></li>
<li class="chapter" data-level="7.2.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#cellranger"><i class="fa fa-check"></i><b>7.2.2</b> Cellranger</a></li>
<li class="chapter" data-level="7.2.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#velocyto-samtools"><i class="fa fa-check"></i><b>7.2.3</b> Velocyto &amp; Samtools</a></li>
<li class="chapter" data-level="7.2.4" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#dropest"><i class="fa fa-check"></i><b>7.2.4</b> <span>Dropest</span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#preprocessing-pipline"><i class="fa fa-check"></i><b>7.3</b> Preprocessing pipline</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#download-and-split-pancreas-e15_5-fastq-data-from-ncbi"><i class="fa fa-check"></i><b>7.3.1</b> 1. Download and split Pancreas E15_5 fastq data from NCBI:</a></li>
<li class="chapter" data-level="7.3.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#run-cellranger-count-referring-to-the-mouse-refdata"><i class="fa fa-check"></i><b>7.3.2</b> 2. Run Cellranger count, referring to the mouse refdata</a></li>
<li class="chapter" data-level="7.3.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#run-velocytodropest."><i class="fa fa-check"></i><b>7.3.3</b> 3. Run Velocyto/Dropest.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">HKU Single-cell Workshop (Modules 4.2 &amp; 5)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="env-pre" class="section level1" number="2">
<h1><span class="header-section-number">Chapter 2</span> Prerequisites</h1>
<div id="quick-summary" class="section level2" number="2.1">
<h2><span class="header-section-number">2.1</span> Quick summary</h2>
<p>For this module, a few tools in Python or command line are needed. If you are already familiar with Unix-based command-line interface, you can prepare your environment with this quick summary, otherwise please see detailed instructions below.</p>
<p>For easier installation and cleaner environment, we recommend using <a href="https://docs.conda.io/en/latest/miniconda.html">conda</a> (a popular environment management tool) with Python 3.7 here.
After having conda installed, you can create an environment with our pre-defined config file <a href="http://ufpr.dl.sourceforge.net/project/sgcellworkshop/sgcell.yml">sgcell.yml</a> (click &amp; download it, and move it to the working directory). Then all you need is this command line:</p>
<pre class="shell"><code>conda env create -f sgcell.yml</code></pre>
<p>Now, you can type <code>conda env list</code>. If the environment named <code>sgcell</code> is listed, it means you configured the environment successfully (now, enjoy this Module).</p>
</div>
<div id="wsl_install" class="section level2" number="2.2">
<h2><span class="header-section-number">2.2</span> (Optional) Install Windows Subsystem for Linux</h2>
<p><strong>Note, this is for Windows users only.</strong> While some required softwares only support Linux or macOS, you could install WSL to use Linux inside Windows.</p>
<p>If you already have Linux or macOS, you can skip this section and jump directly to the section of <code>conda installation</code> <a href="env-pre.html#conda_install_linux">for Linux</a> or <a href="env-pre.html#conda_install_mac">for macOS</a>.</p>
<p>The whole process of installing WSL requires at least 2G disk space. Note that this process was tested on Windows 10 (Version 2004, build 19041.1052).</p>
<div id="what-is-the-windows-subsystem-for-linux-wsl" class="section level3" number="2.2.1">
<h3><span class="header-section-number">2.2.1</span> What is the Windows Subsystem for Linux (WSL)?</h3>
<p>According to the <a href="https://docs.microsoft.com/en-us/windows/wsl/about">Microsoft Docs</a>,</p>
<blockquote>
<p><em>“The Windows Subsystem for Linux lets developers run a GNU/Linux environment – including most command-line tools, utilities, and applications – directly on Windows, unmodified, without the overhead of a traditional virtual machine or dualboot setup.”</em></p>
</blockquote>
</div>
<div id="manual-installation-steps" class="section level3" number="2.2.2">
<h3><span class="header-section-number">2.2.2</span> Manual Installation Steps</h3>
<p><strong>Step 1. Enable required feature in Windows PowerShell</strong></p>
<p>It is necessary to enable the required feature for WSL before installing it.</p>
<ul>
<li>Type <code>powershell</code> in the search box of the Windows taskbar.</li>
<li>Right click <code>Windows PowerShell</code> and select <code>Run as administrator</code>.</li>
</ul>
<p><img src="images/wsl_open_powershell.png" width="45%" style="display: block; margin: auto;" /></p>
<ul>
<li>Type the command below in PowerShell.</li>
</ul>
<pre class="shell"><code>dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart

Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux</code></pre>
<p><img src="images/wsl_enable_feature.png" width="60%" style="display: block; margin: auto;" /></p>
<p><strong>Step 2. Download and install WSL</strong></p>
<p>Microsoft now supports several Linux distributions as WSL, such as Ubuntu, openSUSE, Fedora, etc (<a href="https://docs.microsoft.com/en-us/windows/wsl/install-win10#step-6---install-your-linux-distribution-of-choice">a full list here</a>), among which we choose Ubuntu as an example.</p>
<p>Ubuntu WSL could be freely downloaded and installed through <a href="https://www.microsoft.com/en-us/p/ubuntu-2004-lts/9n6svws3rx71?rtc=1&amp;activetab=pivot:overviewtab">Microsoft Store</a>.</p>
<ul>
<li>Go to the webpage for <a href="https://www.microsoft.com/en-us/p/ubuntu-2004-lts/9n6svws3rx71?rtc=1&amp;activetab=pivot:overviewtab">Ubuntu</a> in Microsoft Store.</li>
<li>Click on the <code>Get</code> button.</li>
<li>Wait for completion of downloading and installation.</li>
<li>Click on the <code>Launch</code> button.</li>
</ul>
<p><img src="images/wsl_download_install.png" width="60%" style="display: block; margin: auto;" /></p>
<p><strong>Step 3. Create a new account for Ubuntu</strong></p>
<p>After successfully installing Ubuntu, a new user account should be created.</p>
<ul>
<li>Type user name and password following the prompts on the screen. Note, it is normal that the password is invisible when you are typing.</li>
</ul>
<p><img src="images/wsl_create_account.png" width="40%" style="display: block; margin: auto;" /></p>
<p><strong>Now Congratulations! You have successfully installed and set up Ubuntu in your Windows System!</strong></p>
<p>Next time you can re-open Ubuntu through the search box of the Windows taskbar.</p>
<p><img src="images/wsl_open_ubuntu.png" width="35%" style="display: block; margin: auto;" /></p>
<p>More information about the usage of WSL can be found at <a href="https://docs.microsoft.com/en-us/windows/wsl">Microsoft Docs</a>.</p>
</div>
</div>
<div id="conda_install" class="section level2" number="2.3">
<h2><span class="header-section-number">2.3</span> Install Conda Environment</h2>
<p>The whole process of installing conda environment requires at least 1G disk space.</p>
<div id="what-is-conda" class="section level3" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> What is conda?</h3>
<p>Conda is part of the Anaconda platform. According to the <a href="https://docs.anaconda.com/#anaconda-individual-edition">Anaconda docs</a>,</p>
<blockquote>
<p><em>“Anaconda Individual Edition is a free, easy-to-install package manager, environment manager, and Python distribution with a collection of 1,500+ open source packages with free community support. Anaconda is platform-agnostic, so you can use it whether you are on Windows, macOS, or Linux.”</em></p>
</blockquote>
<p>For quick installation and configuration, we choose to install <a href="https://docs.conda.io/en/latest/miniconda.html">Miniconda</a> instead of the whole Anaconda.</p>
<blockquote>
<p><em>“Miniconda is a free minimal installer for conda. It is a small, bootstrap version of Anaconda that includes only conda, Python, the packages they depend on, and a small number of other useful packages, including pip, zlib and a few others.”</em></p>
</blockquote>
</div>
<div id="conda_install_windows" class="section level3" number="2.3.2">
<h3><span class="header-section-number">2.3.2</span> Installation on Windows</h3>
<p>Although it is easy to install Miniconda itself on Windows, there are several conda softwares (e.g., bcftools, cellsnp-lite) required for this workshop that are well-supported only on Unix-Like systems. Linux or macOS is essential to use those softwares.</p>
<p>Good news is that Microsoft has provided the Windows Subsystem for Linux (WSL), which enables you to use several Linux distributions on Windows 10. The installation of WSL is easy and details can be found <a href="env-pre.html#wsl_install">here</a>.</p>
<p>After installing WSL, you could follow Section <a href="env-pre.html#conda_install_linux">Installation on Linux</a> to install Miniconda on WSL.</p>
</div>
<div id="conda_install_linux" class="section level3" number="2.3.3">
<h3><span class="header-section-number">2.3.3</span> Installation on Linux</h3>
<p>This section is adapted based on Miniconda installation <a href="https://conda.io/projects/conda/en/latest/user-guide/install/linux.html">guides for Linux</a>. In this section, we select the Miniconda with Python 3.9 as an example.</p>
<p><strong>Step 1. Open Shell</strong></p>
<ul>
<li>Open Linux Shell / WSL Shell (see Section <a href="env-pre.html#wsl_install">Install WSL</a> for details of WSL).</li>
</ul>
<p><strong>Step 2. Download Miniconda installer</strong></p>
<p>The Miniconda installer is a <code>.sh</code> file containing metadata of installation. Run <code>wget &lt;installer_link&gt;</code> to download the installer. e.g.,</p>
<ul>
<li>Type in Shell <code>wget https://repo.anaconda.com/miniconda/Miniconda3-py39_4.9.2-Linux-x86_64.sh</code>.</li>
</ul>
<p><img src="images/conda_download_installer.png" width="65%" style="display: block; margin: auto;" /></p>
<p>You could choose other Miniconda installers based on your platform. A full list of installers can be found <a href="https://docs.conda.io/en/latest/miniconda.html">here</a>.</p>
<p><strong>Step 3. Verify your installer hashes (Optional)</strong></p>
<p>An unmatched hash indicates there were some error during the downloading process. We could simply verify the installer hashes by typing <code>sha256sum &lt;installer_file&gt;</code> in the Shell. e.g.,</p>
<ul>
<li>Type <code>sha256sum Miniconda3-py39_4.9.2-Linux-x86_64.sh</code> and check if the output hash is <code>536817d1b14cb1ada88900f5be51ce0a5e042bae178b5550e62f61e223deae7c</code>.</li>
</ul>
<p><img src="images/conda_verify_hash.PNG" width="65%" style="display: block; margin: auto;" /></p>
<p>A full list of hashes can be found <a href="https://docs.conda.io/en/latest/miniconda.html">here</a>.</p>
<p><strong>Step 4. Initialize Miniconda</strong></p>
<p>Run <code>bash &lt;installer_file&gt;</code> in Shell. e.g.,</p>
<ul>
<li>Type <code>bash Miniconda3-py39_4.9.2-Linux-x86_64.sh</code>.</li>
</ul>
<p><img src="images/conda_init_conda.PNG" width="55%" style="display: block; margin: auto;" /></p>
<ul>
<li>Follow the prompts on the installer screens.</li>
</ul>
<p>When you are asked <code>Do you wish the installer to initialize Miniconda3 by running conda init</code>, we recommend “yes.” When the initialization is done, it looks like,</p>
<p><img src="images/conda_init_conda2.PNG" width="45%" style="display: block; margin: auto;" /></p>
<p><strong>Step 5. Close and re-open Shell</strong></p>
<ul>
<li>Close and then re-open your Shell, to make the changes take effect.</li>
</ul>
<p>The screen should have a new <code>(base)</code> prefix, which looks like,</p>
<p><img src="images/conda_reopen_shell.PNG" width="40%" style="display: block; margin: auto;" /></p>
<p>It means the Shell is now in the <code>base</code> environment of conda.</p>
<p><strong>Step 6. Test your installation</strong></p>
<ul>
<li>Type <code>conda list</code> in Shell.</li>
</ul>
<p>A list of installed packages appears if it has been installed correctly.</p>
<p><img src="images/conda_test_conda.PNG" width="50%" style="display: block; margin: auto;" /></p>
</div>
<div id="conda_install_mac" class="section level3" number="2.3.4">
<h3><span class="header-section-number">2.3.4</span> Installation on macOS</h3>
<p>You could refer to the Section <a href="env-pre.html#conda_install_linux">Installation on Linux</a> as the processes of installing Miniconda on macOS and Linux are quite similar. We briefly list the steps here for your quick reference.</p>
<p><strong>Step 1. Open macOS Terminal</strong></p>
<p><strong>Step 2. Download Miniconda installer</strong></p>
<ul>
<li>Type in Terminal <code>wget https://repo.anaconda.com/miniconda/Miniconda3-py39_4.9.2-MacOSX-x86_64.sh</code>.</li>
</ul>
<p><strong>Step 3. Verify your installer hashes (Optional)</strong></p>
<ul>
<li>Type <code>shasum -a 256 Miniconda3-py39_4.9.2-MacOSX-x86_64.sh</code> in Terminal and check if the output hash is <code>b3bf77cbb81ee235ec6858146a2a84d20f8ecdeb614678030c39baacb5acbed1</code>.</li>
</ul>
<p><strong>Step 4. Initialize Miniconda</strong></p>
<ul>
<li><p>Type in Terminal <code>bash Miniconda3-py39_4.9.2-MacOSX-x86_64.sh</code>.</p></li>
<li><p>Follow the prompts on the installer screens.</p></li>
</ul>
<p>When you are asked <code>Do you wish the installer to initialize Miniconda3 by running conda init</code>, we recommend “yes.”</p>
<p><strong>Step 5. Close and re-open Terminal</strong></p>
<p><strong>Step 6. Test your installation</strong></p>
<ul>
<li>Type <code>conda list</code> in Terminal.</li>
</ul>
<p>A list of installed packages appears if it has been installed correctly.</p>
</div>
</div>
<div id="other-preparation" class="section level2" number="2.4">
<h2><span class="header-section-number">2.4</span> Other Preparation</h2>
<div id="conda-configuration" class="section level3" number="2.4.1">
<h3><span class="header-section-number">2.4.1</span> Conda Configuration</h3>
<p>To simplify the configuration of conda environment, we have pre-compiled a conda environment, named <code>sgcell</code>, containing some required softwares for this workshop. The corresponding metafile <code>sgcell.yml</code> of this environment was exported, based on which we can easily re-build the pre-compiled environment. The <code>sgcell.yml</code> can be
downloaded from <a href="https://sourceforge.net/projects/sgcellworkshop/files/">sgcellworkshop</a> repo on sourceforge.</p>
<p>Assuming you have installed conda and activated any environment in the Shell / Terminal (if not, please install conda based on Section <a href="env-pre.html#conda_install">Install Conda Environment</a> first). Then,</p>
<ul>
<li><p>Open Linux Shell / macOS Terminal / WSL Shell</p></li>
<li><p>Add conda channels,</p></li>
</ul>
<pre class="shell"><code>conda config --add channels bioconda
conda config --add channels conda-forge</code></pre>
<ul>
<li>Re-build the pre-compiled environment (<strong>Note that this can fail due to unstable connection to the server, please re-try for several times</strong>).</li>
</ul>
<pre class="shell"><code>conda env create -f sgcell.yml</code></pre>
<ul>
<li>Activate the new environment <code>sgcell</code>,</li>
</ul>
<pre class="shell"><code>conda activate sgcell</code></pre>
<ul>
<li>Verify if the new environment was installed correctly,</li>
</ul>
<pre class="shell"><code>conda env list</code></pre>
<p>The <code>sgcell</code> sould appear in the listed environments.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="pseudo-time-and-trajectory-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["SingleCell Workshop.pdf", "SingleCell Workshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
