<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 7 Preprocessing of dataset | HKU Single-cell Workshop (Modules 4.2 &amp; 5)</title>
  <meta name="description" content="Cell trajectory inference; Cellular genetic analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 7 Preprocessing of dataset | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 7 Preprocessing of dataset | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  
  <meta name="twitter:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

<meta name="author" content="Yuanhua Lab: Mingze Gao, Rongting Huang, Xianjie Huang, Aaron Kwok, Zhuoxuan Li, Chen Qiao, Yuanhua Huang" />


<meta name="date" content="2021-07-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SingleCell Workshop 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents-of-this-part"><i class="fa fa-check"></i><b>1.1</b> Contents of this part</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#other-modules-of-this-workshop"><i class="fa fa-check"></i><b>1.2</b> Other modules of this workshop</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="env-pre.html"><a href="env-pre.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="2.1" data-path="env-pre.html"><a href="env-pre.html#quick-summary"><i class="fa fa-check"></i><b>2.1</b> Quick summary</a></li>
<li class="chapter" data-level="2.2" data-path="env-pre.html"><a href="env-pre.html#wsl_install"><i class="fa fa-check"></i><b>2.2</b> (Optional) Install Windows Subsystem for Linux</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="env-pre.html"><a href="env-pre.html#what-is-the-windows-subsystem-for-linux-wsl"><i class="fa fa-check"></i><b>2.2.1</b> What is the Windows Subsystem for Linux (WSL)?</a></li>
<li class="chapter" data-level="2.2.2" data-path="env-pre.html"><a href="env-pre.html#manual-installation-steps"><i class="fa fa-check"></i><b>2.2.2</b> Manual Installation Steps</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="env-pre.html"><a href="env-pre.html#conda_install"><i class="fa fa-check"></i><b>2.3</b> Install Conda Environment</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="env-pre.html"><a href="env-pre.html#what-is-conda"><i class="fa fa-check"></i><b>2.3.1</b> What is conda?</a></li>
<li class="chapter" data-level="2.3.2" data-path="env-pre.html"><a href="env-pre.html#conda_install_windows"><i class="fa fa-check"></i><b>2.3.2</b> Installation on Windows</a></li>
<li class="chapter" data-level="2.3.3" data-path="env-pre.html"><a href="env-pre.html#conda_install_linux"><i class="fa fa-check"></i><b>2.3.3</b> Installation on Linux</a></li>
<li class="chapter" data-level="2.3.4" data-path="env-pre.html"><a href="env-pre.html#conda_install_mac"><i class="fa fa-check"></i><b>2.3.4</b> Installation on macOS</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="env-pre.html"><a href="env-pre.html#other-preparation"><i class="fa fa-check"></i><b>2.4</b> Other Preparation</a>
<ul>
<li class="chapter" data-level="2.4.1" data-path="env-pre.html"><a href="env-pre.html#conda-configuration"><i class="fa fa-check"></i><b>2.4.1</b> Conda Configuration</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="pseudo-time-and-trajectory-analysis.html"><a href="pseudo-time-and-trajectory-analysis.html"><i class="fa fa-check"></i><b>3</b> Pseudo-time and trajectory analysis</a></li>
<li class="chapter" data-level="4" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html"><i class="fa fa-check"></i><b>4</b> Trajectory Inference using RNA Velocity</a>
<ul>
<li class="chapter" data-level="4.1" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#use-scvelo-package-within-r"><i class="fa fa-check"></i><b>4.1</b> Use scVelo package within R</a></li>
<li class="chapter" data-level="4.2" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#endocrine-pancreas"><i class="fa fa-check"></i><b>4.2</b> Endocrine Pancreas</a></li>
<li class="chapter" data-level="4.3" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#data-preprocessing"><i class="fa fa-check"></i><b>4.3</b> Data Preprocessing</a></li>
<li class="chapter" data-level="4.4" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#pseudotime-based-on-diffusion-map"><i class="fa fa-check"></i><b>4.4</b> Pseudotime based on Diffusion Map</a></li>
<li class="chapter" data-level="4.5" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#compute-velocity-and-velocity-graph"><i class="fa fa-check"></i><b>4.5</b> Compute velocity and velocity graph</a></li>
<li class="chapter" data-level="4.6" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#diffusion-map-pseudotime-with-velocity"><i class="fa fa-check"></i><b>4.6</b> Diffusion-map Pseudotime with velocity</a></li>
<li class="chapter" data-level="4.7" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#plot-results"><i class="fa fa-check"></i><b>4.7</b> Plot Results</a></li>
<li class="chapter" data-level="4.8" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#interprete-velocity"><i class="fa fa-check"></i><b>4.8</b> Interprete Velocity</a></li>
<li class="chapter" data-level="4.9" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#velocity-in-cycling-progenitors"><i class="fa fa-check"></i><b>4.9</b> Velocity in cycling progenitors</a></li>
<li class="chapter" data-level="4.10" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#dynamical-mode-and-related-analysis"><i class="fa fa-check"></i><b>4.10</b> Dynamical Mode and related analysis</a></li>
<li class="chapter" data-level="4.11" data-path="trajectory-inference-using-rna-velocity.html"><a href="trajectory-inference-using-rna-velocity.html#session-info"><i class="fa fa-check"></i><b>4.11</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html"><i class="fa fa-check"></i><b>5</b> Copy number variation estimation from scRNA-seq</a>
<ul>
<li class="chapter" data-level="5.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#method-infercnv"><i class="fa fa-check"></i><b>5.1</b> Method: inferCNV</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#install-infercnv"><i class="fa fa-check"></i><b>5.1.1</b> install inferCNV</a></li>
<li class="chapter" data-level="5.1.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#getting-started"><i class="fa fa-check"></i><b>5.1.2</b> getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#application-on-tnbc1"><i class="fa fa-check"></i><b>5.2</b> Application on TNBC1</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#data-description"><i class="fa fa-check"></i><b>5.2.1</b> data description</a></li>
<li class="chapter" data-level="5.2.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#run-infercnv"><i class="fa fa-check"></i><b>5.2.2</b> run inferCNV</a></li>
<li class="chapter" data-level="5.2.3" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#infercnv-result"><i class="fa fa-check"></i><b>5.2.3</b> inferCNV result</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#ref"><i class="fa fa-check"></i><b>5.3</b> ref</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><i class="fa fa-check"></i><b>6</b> mtDNA variants for lineage inference from single-cell omics</a>
<ul>
<li class="chapter" data-level="6.1" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html#pileup-with-cellsnp-lite"><i class="fa fa-check"></i><b>6.1</b> Pileup with cellsnp-lite</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html#pileup-without-given-snps"><i class="fa fa-check"></i><b>6.1.1</b> Pileup without given SNPs</a></li>
<li class="chapter" data-level="6.1.2" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html#pileup-with-given-snps"><i class="fa fa-check"></i><b>6.1.2</b> Pileup with given SNPs</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="mtdna-variants-for-lineage-inference-from-single-cell-omics.html"><a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html#clonal-analysis-with-mquad"><i class="fa fa-check"></i><b>6.2</b> Clonal analysis with MQuad</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html"><i class="fa fa-check"></i><b>7</b> Preprocessing of dataset</a>
<ul>
<li class="chapter" data-level="7.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#list-of-packages-aligned-in-pipeline"><i class="fa fa-check"></i><b>7.1</b> List of packages aligned in pipeline:</a></li>
<li class="chapter" data-level="7.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#installation-of-packagessoftwares"><i class="fa fa-check"></i><b>7.2</b> Installation of packages/softwares</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#sratool"><i class="fa fa-check"></i><b>7.2.1</b> SRATool</a></li>
<li class="chapter" data-level="7.2.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#cellranger"><i class="fa fa-check"></i><b>7.2.2</b> Cellranger</a></li>
<li class="chapter" data-level="7.2.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#velocyto-samtools"><i class="fa fa-check"></i><b>7.2.3</b> Velocyto &amp; Samtools</a></li>
<li class="chapter" data-level="7.2.4" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#dropest"><i class="fa fa-check"></i><b>7.2.4</b> <span>Dropest</span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#preprocessing-pipline"><i class="fa fa-check"></i><b>7.3</b> Preprocessing pipline</a>
<ul>
<li class="chapter" data-level="7.3.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#download-and-split-pancreas-e15_5-fastq-data-from-ncbi"><i class="fa fa-check"></i><b>7.3.1</b> 1. Download and split Pancreas E15_5 fastq data from NCBI:</a></li>
<li class="chapter" data-level="7.3.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#run-cellranger-count-referring-to-the-mouse-refdata"><i class="fa fa-check"></i><b>7.3.2</b> 2. Run Cellranger count, referring to the mouse refdata</a></li>
<li class="chapter" data-level="7.3.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#run-velocytodropest."><i class="fa fa-check"></i><b>7.3.3</b> 3. Run Velocyto/Dropest.</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">HKU Single-cell Workshop (Modules 4.2 &amp; 5)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="preprocessing-of-dataset" class="section level1" number="7">
<h1><span class="header-section-number">Chapter 7</span> Preprocessing of dataset</h1>
<p>Author: “Chen Qiao”</p>
<p>Date: “7/2/2021”</p>
<p>In this tutorial, we would take the Pancreas dataset used in scVelo paper for example, show how to retrieve the fastq dataset from NCBI, process using cellranger, velocyto or dropest to obtain the spliced and unspliced count matrices for RNA velocity estimation.</p>
<div id="list-of-packages-aligned-in-pipeline" class="section level2" number="7.1">
<h2><span class="header-section-number">7.1</span> List of packages aligned in pipeline:</h2>
<ol style="list-style-type: decimal">
<li><a href="https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit">SRATool</a>: for downlowding datasets from NCBI, Pancreas dataset E15_5 with SRA id SRR9201794</li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/downloads/latest">Cellranger</a>: for counting, analyzing, QA on fastqs, outputs input for velocyto/dropest</li>
<li>Alternative tools for counting spliced and unspliced reads from Cellranger outputs/
<ul>
<li><a href="http://velocyto.org/velocyto.py/tutorial/cli.html#introduction">Velocyto</a>
<ul>
<li><a href="http://www.htslib.org/">samtools</a>: Velocyto’s dependency pakcage</li>
</ul></li>
<li><a href="https://dropest.readthedocs.io/en/latest/dropest.html">dropest</a></li>
</ul></li>
</ol>
</div>
<div id="installation-of-packagessoftwares" class="section level2" number="7.2">
<h2><span class="header-section-number">7.2</span> Installation of packages/softwares</h2>
<p>All the codes are executed on Linux commandline, lines starting with “#” are comments.
My working path root is “/data/users/cqiao/share/,”
you may change to any foler as you like.</p>
<div id="sratool" class="section level3" number="7.2.1">
<h3><span class="header-section-number">7.2.1</span> SRATool</h3>
<p>Refer to <a href="https://akiomiyao.github.io/ped/sratoolkit/index.html">This tutorial</a> for step 3. software configuration</p>
<pre><code># 1. download software
wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz
# 2. extract package
tar -xzf sratoolkit.tar.gz
# 3. configuration software
./sratoolkit.2.11.0-ubuntu64/vdb-config --interactive
# 4. add software to system path so that we can easily call it
export PATH=$PATH:/data/users/cqiao/share/sratoolkit.2.11.0-ubuntu64/bin/
# 5. test installation, expected to output some sequencing results
fastq-dump --stdout -X 2 SRR390728</code></pre>
</div>
<div id="cellranger" class="section level3" number="7.2.2">
<h3><span class="header-section-number">7.2.2</span> Cellranger</h3>
<pre><code># 1. download software
wget -O cellranger-6.0.2.tar.gz &quot;https://cf.10xgenomics.com/releases/cell-exp/cellranger-6.0.2.tar.gz?Expires=1625079682&amp;Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9jZi4xMHhnZW5vbWljcy5jb20vcmVsZWFzZXMvY2VsbC1leHAvY2VsbHJhbmdlci02LjAuMi50YXIuZ3oiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE2MjUwNzk2ODJ9fX1dfQ__&amp;Signature=Ez05ie1~xJTrUhsc3Q1nZ3JemF5Jc2k3zOOC2PlJiI80TI8YOxM3vVXli90ib1Kl4WgQFRl5~Wp-ce5m2gVAyywGf9juiMk3kW2WGz7xy3DN9nCG-he0rNLvbknDXgkqkj8HeirWyt-EvGgsaqBRZZx7GppGl~MGQOseLTF7DMpJaIm-qwU-72mkVzDy0vKrYVZ~Uiqnlr0M2Hhx14lPwm6QocBuSaKLkmcuZosN4TIS05Z~GPHYt9hQWkykvY4oyd5Y~tyTaMOM8ozVVdI7KgCfc2PmQvhr1bMuhUaEPa3skigzAFjIezbw-fXW3Z8jplwj4lAQ0LybI03ozZr6jg__&amp;Key-Pair-Id=APKAI7S6A5RYOXBWRPDA&quot;
# 2. extract files
tar -xzf cellranger-6.0.2.tar.gz
# 3. add software to system path so that we can easily call it
export PATH=$PATH:/data/users/cqiao/share/cellranger-6.0.2
# 4. download gene refereces (optional saving path: ./cellranger/gtfs/)
cd cellranger-6.0.2
mkdir gtfs
cd gtfs
# human (optional for your needs)
wget https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCh38-2020-A.tar.gz
tar -xzf refdata-gex-GRCh38-2020-A.tar.gz
# mouse (we shall use this for Pancreas dataset)
wget https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-mm10-2020-A.tar.gz
tar -xzf refdata-gex-mm10-2020-A.tar.gz
</code></pre>
</div>
<div id="velocyto-samtools" class="section level3" number="7.2.3">
<h3><span class="header-section-number">7.2.3</span> Velocyto &amp; Samtools</h3>
<div id="samtools" class="section level4" number="7.2.3.1">
<h4><span class="header-section-number">7.2.3.1</span> <a href="http://www.htslib.org/download/">Samtools</a></h4>
<pre><code># 1. download &amp; extract software
wget https://github.com/samtools/samtools/releases/download/1.12/samtools-1.12.tar.bz2
tar -jxvf samtools-1.12.tar.bz2
cd samtools-1.12
# 2. configure build, specify --prefix to locate the root dir
./configure --prefix=/data/users/cqiao/share/samtools
# 3. compile and install software
make
make install
# 4. add software to system path so that we can easily call it
PATH=$PATH:/data/users/cqiao/share/samtools/bin/

# go back to parant folder
cd ..
</code></pre>
</div>
<div id="velocyto" class="section level4" number="7.2.3.2">
<h4><span class="header-section-number">7.2.3.2</span> <a href="http://velocyto.org/velocyto.py/install/index.html">Velocyto</a></h4>
<p>Assume your python packages are managed by conda and pip</p>
<pre><code># install dependent packages, may also use pip
conda install numpy scipy cython numba matplotlib scikit-learn h5py click
# install velocyto
pip install velocyto
</code></pre>
</div>
</div>
<div id="dropest" class="section level3" number="7.2.4">
<h3><span class="header-section-number">7.2.4</span> <a href="https://dropest.readthedocs.io/en/latest/setup.html">Dropest</a></h3>
<p>Install dependent libraries</p>
<p>Install RPackages (in R)</p>
<pre><code>install.packages(c(&quot;Rcpp&quot;,&quot;RcppEigen&quot;, &quot;RInside&quot;, &quot;Matrix&quot;))</code></pre>
<p>Setup a local lib path:</p>
<pre><code>export LOCAL_LIBS=/data/users/cqiao/share/local
mkdir $LOCAL_LIBS
export PATH=$LOCAL_LIBS/bin:$LOCAL_LIBS/usr/local/bin/:$PATH</code></pre>
<p>Install CMake</p>
<pre><code>wget https://cmake.org/files/v3.12/cmake-3.12.0-rc1.tar.gz
tar xvf cmake-3.12.0-rc1.tar.gz
cd cmake-3.12.0-rc1

./bootstrap --prefix=$LOCAL_LIBS
make
make install
cd ..</code></pre>
<p>Zlib</p>
<pre><code>wget https://zlib.net/zlib-1.2.11.tar.gz
tar xvf zlib-1.2.11.tar.gz
cd zlib-1.2.11

./configure --prefix=$LOCAL_LIBS
make
make install
cd ..</code></pre>
<p>BamTools</p>
<pre><code>git clone https://github.com/pezmaster31/bamtools.git
cd bamtools

mkdir build &amp;&amp; cd build
cmake ../
make
make install DESTDIR=$LOCAL_LIBS
cd ..</code></pre>
<p>Bzip2</p>
<pre><code>wget https://www.sourceware.org/pub/bzip2/bzip2-latest.tar.gz
tar -xzvf bzip2-latest.tar.gz
cd bzip2-1.0.8

make -f Makefile-libbz2_so
make install PREFIX=$LOCAL_LIBS
cp -a libbz2.so* $LOCAL_LIBS/lib/
ln -s $LOCAL_LIBS/lib/libbz2.so.1.0 $LOCAL_LIBS/lib/libbz2.so
cd ..</code></pre>
<p>Boost</p>
<pre><code>wget http://sourceforge.net/projects/boost/files/boost/1.60.0/boost_1_60_0.tar.gz
tar xzf boost_1_60_0.tar.gz
cd boost_1_60_0

./bootstrap.sh
./b2 cxxflags=&quot;-std=c++11&quot; include=&quot;$LOCAL_LIBS/include/&quot; search=&quot;$LOCAL_LIBS/lib/&quot; link=shared threading=multi install --prefix=$LOCAL_LIBS
cd ..</code></pre>
<p>DropEst</p>
<pre><code>git clone https://github.com/hms-dbmi/dropEst.git
mkdir dropEst/build
cd dropEst/build
cmake -D CMAKE_INSTALL_PREFIX=&quot;/data/users/cqiao/share/dropesttool&quot; .. &amp;&amp; make
export PATH=$PATH:/data/users/cqiao/share/dropesttool/bin/</code></pre>
</div>
</div>
<div id="preprocessing-pipline" class="section level2" number="7.3">
<h2><span class="header-section-number">7.3</span> Preprocessing pipline</h2>
<div id="download-and-split-pancreas-e15_5-fastq-data-from-ncbi" class="section level3" number="7.3.1">
<h3><span class="header-section-number">7.3.1</span> 1. Download and split Pancreas E15_5 fastq data from NCBI:</h3>
<pre><code>fasterq-dump --split-files SRR9201794</code></pre>
<p>Result would be:</p>
<pre><code>SRR9201794_1.fastq SRR9201794_2.fastq</code></pre>
<p>Rename files for cellranger:</p>
<pre><code>mv SRR9201794_1.fastq SRR9201794_S1_L001_R1_001.fastq
mv SRR9201794_2.fastq SRR9201794_S1_L001_R2_001.fastq</code></pre>
<p>Optinally, move them to a folder <code>/data/users/cqiao/share/E15_5</code></p>
</div>
<div id="run-cellranger-count-referring-to-the-mouse-refdata" class="section level3" number="7.3.2">
<h3><span class="header-section-number">7.3.2</span> 2. Run Cellranger count, referring to the mouse refdata</h3>
<pre><code>cellranger count --id=pancreas_E15_5 \
--fastqs=/data/users/cqiao/share/E15_5 \
--transcriptome=/data/users/cqiao/share/cellranger-6.0.2/gtfs/refdata-gex-mm10-2020-A
</code></pre>
<p><code>--id</code> specifies the output dir, the output root.</p>
</div>
<div id="run-velocytodropest." class="section level3" number="7.3.3">
<h3><span class="header-section-number">7.3.3</span> 3. Run Velocyto/Dropest.</h3>
<p>prepare some variables to store path names, so that afterwards we do not need to use the long names.</p>
<pre><code># Barchodes if cellranger output files: 
gzip -d ./pancreas_E15_5/outs/filtered_feature_bc_matrix/barcodes.tsv.gz
BARCODES=./pancreas_E15_5/outs/filtered_feature_bc_matrix/barcodes.tsv

# Bam file
BAMFILE=./pancreas_E15_5/outs/possorted_genome_bam.bam

# transcriptome reference
GTF=/data/users/cqiao/share/cellranger-6.0.2/gtfs/refdata-gex-mm10-2020-A/genes/genes.gtf</code></pre>
<ul>
<li><p><strong>Velocyto</strong></p>
<pre><code>velocyto run -b $BARCODES -o ./velocyto_res $BAMFILE $GTF
</code></pre>
<p>Results are stored in <code>./velocyto_res/XXX.loom</code></p></li>
<li><p><strong>Dropest</strong></p>
<p>Requires a .xml configuration file specifying the platform settings for sequencing data. Template configure file could be downloaded <a href="https://github.com/kharchenkolab/dropEst/tree/master/configs">here</a>. We shall use 10x.xml for Pancreas E15_5 dataset. Replace line 64 with the path of 10x.xml:</p>
<pre><code>...
       &lt;barcodes_file&gt;./pancreas_E15_5/outs/filtered_feature_bc_matrix/barcodes.tsv&lt;/barcodes_file&gt;
...</code></pre>
<p>Then execute the following code:</p>
<pre><code>$bin/dropest -f -V -g $GTF -c $config_file $BAMFILE</code></pre>
<p>Results are stored in file <code>./pancreas_E15_5/dropEst/cell.counts.matrices.rds</code>.</p></li>
</ul>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mtdna-variants-for-lineage-inference-from-single-cell-omics.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["SingleCell Workshop.pdf", "SingleCell Workshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
