<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Preprocessing of dataset | HKU Single-cell Workshop (Modules 4.2 &amp; 5)</title>
  <meta name="description" content="Cell trajectory inference; Cellular genetic analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Preprocessing of dataset | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Preprocessing of dataset | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  
  <meta name="twitter:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

<meta name="author" content="Yuanhua Lab: Mingze Gao, Rongting Huang, Xianjie Huang, Aaron Kwok, Zhuoxuan Li, Chen Qiao, Yuanhua Huang" />


<meta name="date" content="2021-07-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="copy-number-variation-estimation-from-scrna-seq.html"/>
<link rel="next" href="references.html"/>
<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SingleCell Workshop 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents-of-this-part"><i class="fa fa-check"></i><b>1.1</b> Contents of this part</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#other-modules-of-this-workshop"><i class="fa fa-check"></i><b>1.2</b> Other modules of this workshop</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="env-pre.html"><a href="env-pre.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="2.1" data-path="env-pre.html"><a href="env-pre.html#quick-summary"><i class="fa fa-check"></i><b>2.1</b> Quick summary</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="env-pre.html"><a href="env-pre.html#r-packages"><i class="fa fa-check"></i><b>2.1.1</b> R packages</a></li>
<li class="chapter" data-level="2.1.2" data-path="env-pre.html"><a href="env-pre.html#conda-environment"><i class="fa fa-check"></i><b>2.1.2</b> Conda environment</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html"><i class="fa fa-check"></i><b>3</b> Trajectory inference and RNA velocity</a>
<ul>
<li class="chapter" data-level="3.1" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#introduction-of-trajectory-inference"><i class="fa fa-check"></i><b>3.1</b> Introduction of trajectory inference</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#diffusion-map"><i class="fa fa-check"></i><b>3.1.1</b> Diffusion map</a></li>
<li class="chapter" data-level="3.1.2" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#monocle"><i class="fa fa-check"></i><b>3.1.2</b> Monocle</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#rna-velocity"><i class="fa fa-check"></i><b>3.2</b> RNA velocity</a></li>
<li class="chapter" data-level="3.3" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#rna-kinetics"><i class="fa fa-check"></i><b>3.3</b> RNA kinetics</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#unspliced-rna-indicates-transcriptional-speed"><i class="fa fa-check"></i><b>3.3.1</b> Unspliced RNA indicates transcriptional speed</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#use-scvelo-package-within-r"><i class="fa fa-check"></i><b>3.4</b> Use scVelo package within R</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#environment-and-preprocessing"><i class="fa fa-check"></i><b>3.4.1</b> Environment and preprocessing</a></li>
<li class="chapter" data-level="3.4.2" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#getting-started"><i class="fa fa-check"></i><b>3.4.2</b> Getting started</a></li>
<li class="chapter" data-level="3.4.3" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#data-preprocessing"><i class="fa fa-check"></i><b>3.4.3</b> Data Preprocessing</a></li>
<li class="chapter" data-level="3.4.4" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#pseudotime-based-on-diffusion-map"><i class="fa fa-check"></i><b>3.4.4</b> Pseudotime based on Diffusion Map</a></li>
<li class="chapter" data-level="3.4.5" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#compute-velocity-and-velocity-graph"><i class="fa fa-check"></i><b>3.4.5</b> Compute velocity and velocity graph</a></li>
<li class="chapter" data-level="3.4.6" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#diffusion-map-pseudotime-with-velocity"><i class="fa fa-check"></i><b>3.4.6</b> Diffusion-map Pseudotime with velocity</a></li>
<li class="chapter" data-level="3.4.7" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#plot-results"><i class="fa fa-check"></i><b>3.4.7</b> Plot Results</a></li>
<li class="chapter" data-level="3.4.8" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#interprete-velocity"><i class="fa fa-check"></i><b>3.4.8</b> Interprete Velocity</a></li>
<li class="chapter" data-level="3.4.9" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#velocity-in-cycling-progenitors"><i class="fa fa-check"></i><b>3.4.9</b> Velocity in cycling progenitors</a></li>
<li class="chapter" data-level="3.4.10" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#dynamical-mode-and-related-analysis"><i class="fa fa-check"></i><b>3.4.10</b> Dynamical Mode and related analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html"><i class="fa fa-check"></i><b>4</b> Somatic mutation analysis in single cells</a>
<ul>
<li class="chapter" data-level="4.1" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#choice-of-protocols"><i class="fa fa-check"></i><b>4.1.1</b> Choice of protocols</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#snv-analysis"><i class="fa fa-check"></i><b>4.2</b> SNV analysis</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#call-somatic-variants-and-genotype-single-cells"><i class="fa fa-check"></i><b>4.2.1</b> Call somatic variants and genotype single cells</a></li>
<li class="chapter" data-level="4.2.2" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#visualize-variat-allele-fequency"><i class="fa fa-check"></i><b>4.2.2</b> Visualize variat allele fequency</a></li>
<li class="chapter" data-level="4.2.3" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#use-extra-information-on-clonal-tree"><i class="fa fa-check"></i><b>4.2.3</b> Use extra information on clonal tree</a></li>
<li class="chapter" data-level="4.2.4" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#run-cardelino-for-inferring-clonal-structure"><i class="fa fa-check"></i><b>4.2.4</b> Run cardelino for inferring clonal structure</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#mtsnv-analysis"><i class="fa fa-check"></i><b>4.3</b> mtSNV analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#visualise-allele-frequency"><i class="fa fa-check"></i><b>4.3.1</b> Visualise allele frequency</a></li>
<li class="chapter" data-level="4.3.2" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#infer-clonal-structure-with-mtdna-variants"><i class="fa fa-check"></i><b>4.3.2</b> Infer clonal structure with mtDNA variants</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html"><i class="fa fa-check"></i><b>5</b> Copy number variation estimation from scRNA-seq</a>
<ul>
<li class="chapter" data-level="5.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#method-infercnv"><i class="fa fa-check"></i><b>5.1</b> Method: inferCNV</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#install-infercnv"><i class="fa fa-check"></i><b>5.1.1</b> install inferCNV</a></li>
<li class="chapter" data-level="5.1.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#getting-started-1"><i class="fa fa-check"></i><b>5.1.2</b> getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#application-on-tnbc1"><i class="fa fa-check"></i><b>5.2</b> Application on TNBC1</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#data-description"><i class="fa fa-check"></i><b>5.2.1</b> data description</a></li>
<li class="chapter" data-level="5.2.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#run-infercnv"><i class="fa fa-check"></i><b>5.2.2</b> run inferCNV</a></li>
<li class="chapter" data-level="5.2.3" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#infercnv-result"><i class="fa fa-check"></i><b>5.2.3</b> inferCNV result</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#ref"><i class="fa fa-check"></i><b>5.3</b> ref</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html"><i class="fa fa-check"></i><b>6</b> Preprocessing of dataset</a>
<ul>
<li class="chapter" data-level="6.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#preprocessing-for-rna-velocity"><i class="fa fa-check"></i><b>6.1</b> Preprocessing for RNA Velocity</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#list-of-packages-aligned-in-pipeline"><i class="fa fa-check"></i><b>6.1.1</b> List of packages aligned in pipeline:</a></li>
<li class="chapter" data-level="6.1.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#installation-of-packagessoftwares"><i class="fa fa-check"></i><b>6.1.2</b> Installation of packages/softwares</a></li>
<li class="chapter" data-level="6.1.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#preprocessing-pipline"><i class="fa fa-check"></i><b>6.1.3</b> Preprocessing pipline</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#preprocessing-for-somatic-mutation-analysis"><i class="fa fa-check"></i><b>6.2</b> Preprocessing for somatic mutation analysis</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#pileup-with-cellsnp-lite"><i class="fa fa-check"></i><b>6.2.1</b> Pileup with cellsnp-lite</a></li>
<li class="chapter" data-level="6.2.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#clonal-analysis-with-mquad"><i class="fa fa-check"></i><b>6.2.2</b> Clonal analysis with MQuad</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#wsl_install"><i class="fa fa-check"></i><b>6.3</b> (Optional) Install Windows Subsystem for Linux</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#what-is-the-windows-subsystem-for-linux-wsl"><i class="fa fa-check"></i><b>6.3.1</b> What is the Windows Subsystem for Linux (WSL)?</a></li>
<li class="chapter" data-level="6.3.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#manual-installation-steps"><i class="fa fa-check"></i><b>6.3.2</b> Manual Installation Steps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">HKU Single-cell Workshop (Modules 4.2 &amp; 5)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="preprocessing-of-dataset" class="section level1" number="6">
<h1><span class="header-section-number">Chapter 6</span> Preprocessing of dataset</h1>
<div id="preprocessing-for-rna-velocity" class="section level2" number="6.1">
<h2><span class="header-section-number">6.1</span> Preprocessing for RNA Velocity</h2>
<p>Author: “Chen Qiao”</p>
<p>Date: “7/2/2021”</p>
<p>In this tutorial, we would take the Pancreas dataset used in scVelo paper for example, show how to retrieve the fastq dataset from NCBI, process using cellranger, velocyto or dropest to obtain the spliced and unspliced count matrices for RNA velocity estimation.</p>
<div id="list-of-packages-aligned-in-pipeline" class="section level3" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> List of packages aligned in pipeline:</h3>
<ol style="list-style-type: decimal">
<li><a href="https://github.com/ncbi/sra-tools/wiki/02.-Installing-SRA-Toolkit">SRATool</a>: for downlowding datasets from NCBI, Pancreas dataset E15_5 with SRA id SRR9201794</li>
<li><a href="https://support.10xgenomics.com/single-cell-gene-expression/software/downloads/latest">Cellranger</a>: for counting, analyzing, QA on fastqs, outputs input for velocyto/dropest</li>
<li>Alternative tools for counting spliced and unspliced reads from Cellranger outputs/
<ul>
<li><a href="http://velocyto.org/velocyto.py/tutorial/cli.html#introduction">Velocyto</a>
<ul>
<li><a href="http://www.htslib.org/">samtools</a>: Velocyto’s dependency pakcage</li>
</ul></li>
<li><a href="https://dropest.readthedocs.io/en/latest/dropest.html">dropest</a></li>
</ul></li>
</ol>
</div>
<div id="installation-of-packagessoftwares" class="section level3" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> Installation of packages/softwares</h3>
<p>All the codes are executed on Linux commandline, lines starting with “#” are comments.
My working path root is “/data/users/cqiao/share/”,
you may change to any foler as you like.</p>
<div id="sratool" class="section level4" number="6.1.2.1">
<h4><span class="header-section-number">6.1.2.1</span> SRATool</h4>
<p>Refer to <a href="https://akiomiyao.github.io/ped/sratoolkit/index.html">This tutorial</a> for step 3. software configuration</p>
<pre><code># 1. download software
wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/current/sratoolkit.current-ubuntu64.tar.gz
# 2. extract package
tar -xzf sratoolkit.tar.gz
# 3. configuration software
./sratoolkit.2.11.0-ubuntu64/vdb-config --interactive
# 4. add software to system path so that we can easily call it
export PATH=$PATH:/data/users/cqiao/share/sratoolkit.2.11.0-ubuntu64/bin/
# 5. test installation, expected to output some sequencing results
fastq-dump --stdout -X 2 SRR390728</code></pre>
</div>
<div id="cellranger" class="section level4" number="6.1.2.2">
<h4><span class="header-section-number">6.1.2.2</span> Cellranger</h4>
<pre><code># 1. download software
wget -O cellranger-6.0.2.tar.gz &quot;https://cf.10xgenomics.com/releases/cell-exp/cellranger-6.0.2.tar.gz?Expires=1625079682&amp;Policy=eyJTdGF0ZW1lbnQiOlt7IlJlc291cmNlIjoiaHR0cHM6Ly9jZi4xMHhnZW5vbWljcy5jb20vcmVsZWFzZXMvY2VsbC1leHAvY2VsbHJhbmdlci02LjAuMi50YXIuZ3oiLCJDb25kaXRpb24iOnsiRGF0ZUxlc3NUaGFuIjp7IkFXUzpFcG9jaFRpbWUiOjE2MjUwNzk2ODJ9fX1dfQ__&amp;Signature=Ez05ie1~xJTrUhsc3Q1nZ3JemF5Jc2k3zOOC2PlJiI80TI8YOxM3vVXli90ib1Kl4WgQFRl5~Wp-ce5m2gVAyywGf9juiMk3kW2WGz7xy3DN9nCG-he0rNLvbknDXgkqkj8HeirWyt-EvGgsaqBRZZx7GppGl~MGQOseLTF7DMpJaIm-qwU-72mkVzDy0vKrYVZ~Uiqnlr0M2Hhx14lPwm6QocBuSaKLkmcuZosN4TIS05Z~GPHYt9hQWkykvY4oyd5Y~tyTaMOM8ozVVdI7KgCfc2PmQvhr1bMuhUaEPa3skigzAFjIezbw-fXW3Z8jplwj4lAQ0LybI03ozZr6jg__&amp;Key-Pair-Id=APKAI7S6A5RYOXBWRPDA&quot;
# 2. extract files
tar -xzf cellranger-6.0.2.tar.gz
# 3. add software to system path so that we can easily call it
export PATH=$PATH:/data/users/cqiao/share/cellranger-6.0.2
# 4. download gene refereces (optional saving path: ./cellranger/gtfs/)
cd cellranger-6.0.2
mkdir gtfs
cd gtfs
# human (optional for your needs)
wget https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-GRCh38-2020-A.tar.gz
tar -xzf refdata-gex-GRCh38-2020-A.tar.gz
# mouse (we shall use this for Pancreas dataset)
wget https://cf.10xgenomics.com/supp/cell-exp/refdata-gex-mm10-2020-A.tar.gz
tar -xzf refdata-gex-mm10-2020-A.tar.gz
</code></pre>
</div>
<div id="velocyto-samtools" class="section level4" number="6.1.2.3">
<h4><span class="header-section-number">6.1.2.3</span> Velocyto &amp; Samtools</h4>
<div id="samtools" class="section level5" number="6.1.2.3.1">
<h5><span class="header-section-number">6.1.2.3.1</span> <a href="http://www.htslib.org/download/">Samtools</a></h5>
<pre><code># 1. download &amp; extract software
wget https://github.com/samtools/samtools/releases/download/1.12/samtools-1.12.tar.bz2
tar -jxvf samtools-1.12.tar.bz2
cd samtools-1.12
# 2. configure build, specify --prefix to locate the root dir
./configure --prefix=/data/users/cqiao/share/samtools
# 3. compile and install software
make
make install
# 4. add software to system path so that we can easily call it
PATH=$PATH:/data/users/cqiao/share/samtools/bin/

# go back to parant folder
cd ..
</code></pre>
</div>
<div id="velocyto" class="section level5" number="6.1.2.3.2">
<h5><span class="header-section-number">6.1.2.3.2</span> <a href="http://velocyto.org/velocyto.py/install/index.html">Velocyto</a></h5>
<p>Assume your python packages are managed by conda and pip</p>
<pre><code># install dependent packages, may also use pip
conda install numpy scipy cython numba matplotlib scikit-learn h5py click
# install velocyto
pip install velocyto
</code></pre>
</div>
</div>
<div id="dropest" class="section level4" number="6.1.2.4">
<h4><span class="header-section-number">6.1.2.4</span> <a href="https://dropest.readthedocs.io/en/latest/setup.html">Dropest</a></h4>
<p>Install dependent libraries</p>
<p>Install RPackages (in R)</p>
<pre><code>install.packages(c(&quot;Rcpp&quot;,&quot;RcppEigen&quot;, &quot;RInside&quot;, &quot;Matrix&quot;))</code></pre>
<p>Setup a local lib path:</p>
<pre><code>export LOCAL_LIBS=/data/users/cqiao/share/local
mkdir $LOCAL_LIBS
export PATH=$LOCAL_LIBS/bin:$LOCAL_LIBS/usr/local/bin/:$PATH</code></pre>
<p>Install CMake</p>
<pre><code>wget https://cmake.org/files/v3.12/cmake-3.12.0-rc1.tar.gz
tar xvf cmake-3.12.0-rc1.tar.gz
cd cmake-3.12.0-rc1

./bootstrap --prefix=$LOCAL_LIBS
make
make install
cd ..</code></pre>
<p>Zlib</p>
<pre><code>wget https://zlib.net/zlib-1.2.11.tar.gz
tar xvf zlib-1.2.11.tar.gz
cd zlib-1.2.11

./configure --prefix=$LOCAL_LIBS
make
make install
cd ..</code></pre>
<p>BamTools</p>
<pre><code>git clone https://github.com/pezmaster31/bamtools.git
cd bamtools

mkdir build &amp;&amp; cd build
cmake ../
make
make install DESTDIR=$LOCAL_LIBS
cd ..</code></pre>
<p>Bzip2</p>
<pre><code>wget https://www.sourceware.org/pub/bzip2/bzip2-latest.tar.gz
tar -xzvf bzip2-latest.tar.gz
cd bzip2-1.0.8

make -f Makefile-libbz2_so
make install PREFIX=$LOCAL_LIBS
cp -a libbz2.so* $LOCAL_LIBS/lib/
ln -s $LOCAL_LIBS/lib/libbz2.so.1.0 $LOCAL_LIBS/lib/libbz2.so
cd ..</code></pre>
<p>Boost</p>
<pre><code>wget http://sourceforge.net/projects/boost/files/boost/1.60.0/boost_1_60_0.tar.gz
tar xzf boost_1_60_0.tar.gz
cd boost_1_60_0

./bootstrap.sh
./b2 cxxflags=&quot;-std=c++11&quot; include=&quot;$LOCAL_LIBS/include/&quot; search=&quot;$LOCAL_LIBS/lib/&quot; link=shared threading=multi install --prefix=$LOCAL_LIBS
cd ..</code></pre>
<p>DropEst</p>
<pre><code>git clone https://github.com/hms-dbmi/dropEst.git
mkdir dropEst/build
cd dropEst/build
cmake -D CMAKE_INSTALL_PREFIX=&quot;/data/users/cqiao/share/dropesttool&quot; .. &amp;&amp; make
export PATH=$PATH:/data/users/cqiao/share/dropesttool/bin/</code></pre>
</div>
</div>
<div id="preprocessing-pipline" class="section level3" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> Preprocessing pipline</h3>
<div id="download-and-split-pancreas-e15_5-fastq-data-from-ncbi" class="section level4" number="6.1.3.1">
<h4><span class="header-section-number">6.1.3.1</span> 1. Download and split Pancreas E15_5 fastq data from NCBI:</h4>
<pre><code>fasterq-dump --split-files SRR9201794</code></pre>
<p>Result would be:</p>
<pre><code>SRR9201794_1.fastq SRR9201794_2.fastq</code></pre>
<p>Rename files for cellranger:</p>
<pre><code>mv SRR9201794_1.fastq SRR9201794_S1_L001_R1_001.fastq
mv SRR9201794_2.fastq SRR9201794_S1_L001_R2_001.fastq</code></pre>
<p>Optinally, move them to a folder <code>/data/users/cqiao/share/E15_5</code></p>
</div>
<div id="run-cellranger-count-referring-to-the-mouse-refdata" class="section level4" number="6.1.3.2">
<h4><span class="header-section-number">6.1.3.2</span> 2. Run Cellranger count, referring to the mouse refdata</h4>
<pre><code>cellranger count --id=pancreas_E15_5 \
--fastqs=/data/users/cqiao/share/E15_5 \
--transcriptome=/data/users/cqiao/share/cellranger-6.0.2/gtfs/refdata-gex-mm10-2020-A
</code></pre>
<p><code>--id</code> specifies the output dir, the output root.</p>
</div>
<div id="run-velocytodropest." class="section level4" number="6.1.3.3">
<h4><span class="header-section-number">6.1.3.3</span> 3. Run Velocyto/Dropest.</h4>
<p>prepare some variables to store path names, so that afterwards we do not need to use the long names.</p>
<pre><code># Barchodes if cellranger output files: 
gzip -d ./pancreas_E15_5/outs/filtered_feature_bc_matrix/barcodes.tsv.gz
BARCODES=./pancreas_E15_5/outs/filtered_feature_bc_matrix/barcodes.tsv

# Bam file
BAMFILE=./pancreas_E15_5/outs/possorted_genome_bam.bam

# transcriptome reference
GTF=/data/users/cqiao/share/cellranger-6.0.2/gtfs/refdata-gex-mm10-2020-A/genes/genes.gtf</code></pre>
<ul>
<li><p><strong>Velocyto</strong></p>
<pre><code>velocyto run -b $BARCODES -o ./velocyto_res $BAMFILE $GTF
</code></pre>
<p>Results are stored in <code>./velocyto_res/XXX.loom</code></p></li>
<li><p><strong>Dropest</strong></p>
<p>Requires a .xml configuration file specifying the platform settings for sequencing data. Template configure file could be downloaded <a href="https://github.com/kharchenkolab/dropEst/tree/master/configs">here</a>. We shall use 10x.xml for Pancreas E15_5 dataset. Replace line 64 with the path of 10x.xml:</p>
<pre><code>...
       &lt;barcodes_file&gt;./pancreas_E15_5/outs/filtered_feature_bc_matrix/barcodes.tsv&lt;/barcodes_file&gt;
...</code></pre>
<p>Then execute the following code:</p>
<pre><code>$bin/dropest -f -V -g $GTF -c $config_file $BAMFILE</code></pre>
<p>Results are stored in file <code>./pancreas_E15_5/dropEst/cell.counts.matrices.rds</code>.</p></li>
</ul>
</div>
</div>
</div>
<div id="preprocessing-for-somatic-mutation-analysis" class="section level2" number="6.2">
<h2><span class="header-section-number">6.2</span> Preprocessing for somatic mutation analysis</h2>
<div id="pileup-with-cellsnp-lite" class="section level3" number="6.2.1">
<h3><span class="header-section-number">6.2.1</span> Pileup with cellsnp-lite</h3>
<p>Cellsnp-lite is designed to perform efficient pileup and genotyping for both bulk and single cell sequencing data. It could be easily installed via <a href="#conda_install">conda</a> with <code>conda install -c bioconda -c conda-forge cellsnp-lite</code>. Note that, cellsnp-lite does not support Windows, but you can use it with the help of the <code>Windows Subsystem for Linux (WSL)</code>. Details about WSL can be found in section below.</p>
<p>We use a SMART-seq2 dataset as an example, which were packed into <code>joxm.bam.all.77.tar.gz</code> (77 bam files, ~3.5G in total; <a href="https://www.nature.com/articles/s41592-020-0766-3">McCarthy et al, 2020</a>).</p>
<p>First, please download <code>joxm.bam.all.77.tar.gz</code>, <code>joxm.hg19.bam.lst</code> and <code>joxm.sample.lst</code> from <a href="https://sourceforge.net/projects/sgcellworkshop/files/mtDNA_analysis/">sgcellworkshop</a> repo on sourceforge. Put the three files in the same directory. Then unpack the bam files with command <code>tar zxvf joxm.bam.all.77.tar.gz</code>. The bam files, together with the <code>.bai</code> files, should be in the <code>sort/</code> directory.</p>
<div id="pileup-without-given-snps" class="section level4" number="6.2.1.1">
<h4><span class="header-section-number">6.2.1.1</span> Pileup without given SNPs</h4>
<p>In this example, we use cellsnp-lite to pileup chrM on the bam files without a given SNP list (in denovo mode), whose output could be used as inputs of MQuad model for mitochondrial clone analysis.</p>
<p>Note that the pileup results <code>joxm.hg19.cellsnp.mode2b.tar.gz</code> can be downloaded from <a href="https://sourceforge.net/projects/sgcellworkshop/files/mtDNA_analysis/">sgcellworkshop</a> repo on sourceforge and then unpacked with command <code>tar zxvf joxm.hg19.cellsnp.mode2b.tar.gz</code>.</p>
<p>The command line to generate the results is,</p>
<pre class="shell"><code>cellsnp-lite      \
  -S ./joxm.hg19.bam.lst    \
  -i ./joxm.sample.lst      \
  -O ./cellsnp2b            \
  --chrom MT        \
  --cellTAG None    \
  --UMItag None     \
  --minCOUNT 20     \
  --minMAF 0        \
  --genotype        \
  --gzip</code></pre>
</div>
<div id="pileup-with-given-snps" class="section level4" number="6.2.1.2">
<h4><span class="header-section-number">6.2.1.2</span> Pileup with given SNPs</h4>
<p>This time, we use cellsnp-lite to pileup given SNPs on the bam files. The 112 SNPs, stored in <code>joxm.hg19.somatic.sites.vcf</code>, were somatic SNPs called in (<a href="https://www.nature.com/articles/s41592-020-0766-3">McCarthy et al, 2020</a>) and can be downloaded from <a href="https://sourceforge.net/projects/sgcellworkshop/files/mtDNA_analysis/">sgcellworkshop</a> repo.</p>
<p>Note that the pileup results <code>joxm.hg19.somatic.cellsnp.mode1b.tar.gz</code> can be downloaded from <a href="https://sourceforge.net/projects/sgcellworkshop/files/mtDNA_analysis/">sgcellworkshop</a> repo on sourceforge and then unpacked with command <code>tar zxvf joxm.hg19.somatic.cellsnp.mode1b.tar.gz</code>.</p>
<p>The command line to generate the results is,</p>
<pre class="shell"><code>cellsnp-lite      \
  -S ./joxm.hg19.bam.lst    \
  -i ./joxm.sample.lst      \
  -O ./cellsnp1b            \
  -R ./joxm.hg19.somatic.sites.vcf \
  --cellTAG None    \
  --UMItag None     \
  --minCOUNT 0     \
  --minMAF 0        \
  -p 4              \
  --genotype        \
  --gzip</code></pre>
</div>
</div>
<div id="clonal-analysis-with-mquad" class="section level3" number="6.2.2">
<h3><span class="header-section-number">6.2.2</span> Clonal analysis with MQuad</h3>
</div>
</div>
<div id="wsl_install" class="section level2" number="6.3">
<h2><span class="header-section-number">6.3</span> (Optional) Install Windows Subsystem for Linux</h2>
<p><strong>Note, this is for Windows users only.</strong> While some required softwares only support Linux or macOS, you could install WSL to use Linux inside Windows.</p>
<p>The whole process of installing WSL requires at least 2G disk space. Note that this process was tested on Windows 10 (Version 2004, build 19041.1052).</p>
<div id="what-is-the-windows-subsystem-for-linux-wsl" class="section level3" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> What is the Windows Subsystem for Linux (WSL)?</h3>
<p>According to the <a href="https://docs.microsoft.com/en-us/windows/wsl/about">Microsoft Docs</a>,</p>
<blockquote>
<p><em>“The Windows Subsystem for Linux lets developers run a GNU/Linux environment – including most command-line tools, utilities, and applications – directly on Windows, unmodified, without the overhead of a traditional virtual machine or dualboot setup.”</em></p>
</blockquote>
</div>
<div id="manual-installation-steps" class="section level3" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Manual Installation Steps</h3>
<p><strong>Step 1. Enable required feature in Windows PowerShell</strong></p>
<p>It is necessary to enable the required feature for WSL before installing it.</p>
<ul>
<li>Type <code>powershell</code> in the search box of the Windows taskbar.</li>
<li>Right click <code>Windows PowerShell</code> and select <code>Run as administrator</code>.</li>
</ul>
<p><img src="images/wsl_open_powershell.png" width="45%" style="display: block; margin: auto;" /></p>
<ul>
<li>Type the command below in PowerShell.</li>
</ul>
<pre class="shell"><code>dism.exe /online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all /norestart

Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux</code></pre>
<p><img src="images/wsl_enable_feature.png" width="60%" style="display: block; margin: auto;" /></p>
<p><strong>Step 2. Download and install WSL</strong></p>
<p>Microsoft now supports several Linux distributions as WSL, such as Ubuntu, openSUSE, Fedora, etc (<a href="https://docs.microsoft.com/en-us/windows/wsl/install-win10#step-6---install-your-linux-distribution-of-choice">a full list here</a>), among which we choose Ubuntu as an example.</p>
<p>Ubuntu WSL could be freely downloaded and installed through <a href="https://www.microsoft.com/en-us/p/ubuntu-2004-lts/9n6svws3rx71?rtc=1&amp;activetab=pivot:overviewtab">Microsoft Store</a>.</p>
<ul>
<li>Go to the webpage for <a href="https://www.microsoft.com/en-us/p/ubuntu-2004-lts/9n6svws3rx71?rtc=1&amp;activetab=pivot:overviewtab">Ubuntu</a> in Microsoft Store.</li>
<li>Click on the <code>Get</code> button.</li>
<li>Wait for completion of downloading and installation.</li>
<li>Click on the <code>Launch</code> button.</li>
</ul>
<p><img src="images/wsl_download_install.png" width="60%" style="display: block; margin: auto;" /></p>
<p><strong>Step 3. Create a new account for Ubuntu</strong></p>
<p>After successfully installing Ubuntu, a new user account should be created.</p>
<ul>
<li>Type user name and password following the prompts on the screen. Note, it is normal that the password is invisible when you are typing.</li>
</ul>
<p><img src="images/wsl_create_account.png" width="40%" style="display: block; margin: auto;" /></p>
<p><strong>Now Congratulations! You have successfully installed and set up Ubuntu in your Windows System!</strong></p>
<p>Next time you can re-open Ubuntu through the search box of the Windows taskbar.</p>
<p><img src="images/wsl_open_ubuntu.png" width="35%" style="display: block; margin: auto;" /></p>
<p>More information about the usage of WSL can be found at <a href="https://docs.microsoft.com/en-us/windows/wsl">Microsoft Docs</a>.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="copy-number-variation-estimation-from-scrna-seq.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="references.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["SingleCell Workshop.pdf", "SingleCell Workshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
