<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>References | HKU Single-cell Workshop (Modules 4.2 &amp; 5)</title>
  <meta name="description" content="Cell trajectory inference; Cellular genetic analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="References | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="References | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  
  <meta name="twitter:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

<meta name="author" content="Yuanhua Lab: Mingze Gao, Rongting Huang, Xianjie Huang, Aaron Kwok, Zhuoxuan Li, Chen Qiao, Yuanhua Huang contact: yuanhua@hku.hk" />


<meta name="date" content="2021-07-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="preprocess.html"/>

<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SingleCell Workshop 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents-of-this-part"><i class="fa fa-check"></i><b>1.1</b> Contents of this part</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#other-modules-of-this-workshop"><i class="fa fa-check"></i><b>1.2</b> Other modules of this workshop</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="env-pre.html"><a href="env-pre.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="2.1" data-path="env-pre.html"><a href="env-pre.html#r-packages"><i class="fa fa-check"></i><b>2.1</b> R packages</a></li>
<li class="chapter" data-level="2.2" data-path="env-pre.html"><a href="env-pre.html#python_packages"><i class="fa fa-check"></i><b>2.2</b> Python packages</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="env-pre.html"><a href="env-pre.html#conda_install_windows"><i class="fa fa-check"></i><b>2.2.1</b> Installation on Windows</a></li>
<li class="chapter" data-level="2.2.2" data-path="env-pre.html"><a href="env-pre.html#conda_install_linux"><i class="fa fa-check"></i><b>2.2.2</b> Installation on Linux</a></li>
<li class="chapter" data-level="2.2.3" data-path="env-pre.html"><a href="env-pre.html#conda_install_mac"><i class="fa fa-check"></i><b>2.2.3</b> Installation on macOS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="RNA-velocity.html"><a href="RNA-velocity.html"><i class="fa fa-check"></i><b>3</b> Trajectory inference and RNA velocity</a>
<ul>
<li class="chapter" data-level="3.1" data-path="RNA-velocity.html"><a href="RNA-velocity.html#trajectory-inference"><i class="fa fa-check"></i><b>3.1</b> Trajectory inference</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="RNA-velocity.html"><a href="RNA-velocity.html#diffusion-map"><i class="fa fa-check"></i><b>3.1.1</b> Diffusion map</a></li>
<li class="chapter" data-level="3.1.2" data-path="RNA-velocity.html"><a href="RNA-velocity.html#monocle"><i class="fa fa-check"></i><b>3.1.2</b> Monocle</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="RNA-velocity.html"><a href="RNA-velocity.html#rna-velocity"><i class="fa fa-check"></i><b>3.2</b> RNA velocity</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="RNA-velocity.html"><a href="RNA-velocity.html#rna-kinetics"><i class="fa fa-check"></i><b>3.2.1</b> RNA kinetics</a></li>
<li class="chapter" data-level="3.2.2" data-path="RNA-velocity.html"><a href="RNA-velocity.html#unspliced_RNA"><i class="fa fa-check"></i><b>3.2.2</b> Unspliced RNA indicates transcriptional speed</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="RNA-velocity.html"><a href="RNA-velocity.html#scvelo-in-r"><i class="fa fa-check"></i><b>3.3</b> scVelo in R</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="RNA-velocity.html"><a href="RNA-velocity.html#getting-started"><i class="fa fa-check"></i><b>3.3.1</b> Getting started</a></li>
<li class="chapter" data-level="3.3.2" data-path="RNA-velocity.html"><a href="RNA-velocity.html#data-preprocessing"><i class="fa fa-check"></i><b>3.3.2</b> Data Preprocessing</a></li>
<li class="chapter" data-level="3.3.3" data-path="RNA-velocity.html"><a href="RNA-velocity.html#diffusion-pseudotime"><i class="fa fa-check"></i><b>3.3.3</b> Diffusion pseudotime</a></li>
<li class="chapter" data-level="3.3.4" data-path="RNA-velocity.html"><a href="RNA-velocity.html#compute-velocity-and-velocity-graph"><i class="fa fa-check"></i><b>3.3.4</b> Compute velocity and velocity graph</a></li>
<li class="chapter" data-level="3.3.5" data-path="RNA-velocity.html"><a href="RNA-velocity.html#diffusion-pseudotime-with-velocity"><i class="fa fa-check"></i><b>3.3.5</b> Diffusion pseudotime with velocity</a></li>
<li class="chapter" data-level="3.3.6" data-path="RNA-velocity.html"><a href="RNA-velocity.html#visualise-velocity-based-trajectory"><i class="fa fa-check"></i><b>3.3.6</b> Visualise velocity-based trajectory</a></li>
<li class="chapter" data-level="3.3.7" data-path="RNA-velocity.html"><a href="RNA-velocity.html#interprete-velocity"><i class="fa fa-check"></i><b>3.3.7</b> Interprete Velocity</a></li>
<li class="chapter" data-level="3.3.8" data-path="RNA-velocity.html"><a href="RNA-velocity.html#velocity-in-cycling-progenitors"><i class="fa fa-check"></i><b>3.3.8</b> Velocity in cycling progenitors</a></li>
<li class="chapter" data-level="3.3.9" data-path="RNA-velocity.html"><a href="RNA-velocity.html#dynamical-mode-and-related-analysis"><i class="fa fa-check"></i><b>3.3.9</b> Dynamical Mode and related analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="RNA-velocity.html"><a href="RNA-velocity.html#last-notes"><i class="fa fa-check"></i><b>3.4</b> Last notes</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="SNV-analysis.html"><a href="SNV-analysis.html"><i class="fa fa-check"></i><b>4</b> Somatic mutation analysis in single cells</a>
<ul>
<li class="chapter" data-level="4.1" data-path="SNV-analysis.html"><a href="SNV-analysis.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="SNV-analysis.html"><a href="SNV-analysis.html#choice-of-protocols"><i class="fa fa-check"></i><b>4.1.1</b> Choice of protocols</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="SNV-analysis.html"><a href="SNV-analysis.html#snv-analysis"><i class="fa fa-check"></i><b>4.2</b> SNV analysis</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="SNV-analysis.html"><a href="SNV-analysis.html#call-somatic-variants-and-genotype-single-cells"><i class="fa fa-check"></i><b>4.2.1</b> Call somatic variants and genotype single cells</a></li>
<li class="chapter" data-level="4.2.2" data-path="SNV-analysis.html"><a href="SNV-analysis.html#visualize-variat-allele-fequency"><i class="fa fa-check"></i><b>4.2.2</b> Visualize variat allele fequency</a></li>
<li class="chapter" data-level="4.2.3" data-path="SNV-analysis.html"><a href="SNV-analysis.html#use-extra-information-on-clonal-tree"><i class="fa fa-check"></i><b>4.2.3</b> Use extra information on clonal tree</a></li>
<li class="chapter" data-level="4.2.4" data-path="SNV-analysis.html"><a href="SNV-analysis.html#run-cardelino-for-inferring-clonal-structure"><i class="fa fa-check"></i><b>4.2.4</b> Run cardelino for inferring clonal structure</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="SNV-analysis.html"><a href="SNV-analysis.html#mtsnv-analysis"><i class="fa fa-check"></i><b>4.3</b> mtSNV analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="SNV-analysis.html"><a href="SNV-analysis.html#visualise-allele-frequency"><i class="fa fa-check"></i><b>4.3.1</b> Visualise allele frequency</a></li>
<li class="chapter" data-level="4.3.2" data-path="SNV-analysis.html"><a href="SNV-analysis.html#infer-clonal-structure-with-mtdna-variants"><i class="fa fa-check"></i><b>4.3.2</b> Infer clonal structure with mtDNA variants</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="SNV-analysis.html"><a href="SNV-analysis.html#last-notes-1"><i class="fa fa-check"></i><b>4.4</b> Last notes</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="CNV-analysis.html"><a href="CNV-analysis.html"><i class="fa fa-check"></i><b>5</b> Copy number variation estimation from scRNA-seq</a>
<ul>
<li class="chapter" data-level="5.1" data-path="CNV-analysis.html"><a href="CNV-analysis.html#introduction-2"><i class="fa fa-check"></i><b>5.1</b> Introduction</a></li>
<li class="chapter" data-level="5.2" data-path="CNV-analysis.html"><a href="CNV-analysis.html#infercnv-and-example"><i class="fa fa-check"></i><b>5.2</b> inferCNV and example</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="CNV-analysis.html"><a href="CNV-analysis.html#install-infercnv"><i class="fa fa-check"></i><b>5.2.1</b> install inferCNV</a></li>
<li class="chapter" data-level="5.2.2" data-path="CNV-analysis.html"><a href="CNV-analysis.html#getting-started-1"><i class="fa fa-check"></i><b>5.2.2</b> getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="CNV-analysis.html"><a href="CNV-analysis.html#application-on-tnbc1"><i class="fa fa-check"></i><b>5.3</b> Application on TNBC1</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="CNV-analysis.html"><a href="CNV-analysis.html#data-description"><i class="fa fa-check"></i><b>5.3.1</b> data description</a></li>
<li class="chapter" data-level="5.3.2" data-path="CNV-analysis.html"><a href="CNV-analysis.html#run-infercnv"><i class="fa fa-check"></i><b>5.3.2</b> run inferCNV</a></li>
<li class="chapter" data-level="5.3.3" data-path="CNV-analysis.html"><a href="CNV-analysis.html#infercnv-result"><i class="fa fa-check"></i><b>5.3.3</b> inferCNV result</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="CNV-analysis.html"><a href="CNV-analysis.html#last-notes-2"><i class="fa fa-check"></i><b>5.4</b> Last notes</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="preprocess.html"><a href="preprocess.html"><i class="fa fa-check"></i><b>6</b> Preprocessing of dataset</a>
<ul>
<li class="chapter" data-level="6.1" data-path="preprocess.html"><a href="preprocess.html#preprocessing-for-rna-velocity"><i class="fa fa-check"></i><b>6.1</b> Preprocessing for RNA Velocity</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="preprocess.html"><a href="preprocess.html#list-of-packages-aligned-in-pipeline"><i class="fa fa-check"></i><b>6.1.1</b> List of packages aligned in pipeline:</a></li>
<li class="chapter" data-level="6.1.2" data-path="preprocess.html"><a href="preprocess.html#installation-of-packagessoftwares"><i class="fa fa-check"></i><b>6.1.2</b> Installation of packages/softwares</a></li>
<li class="chapter" data-level="6.1.3" data-path="preprocess.html"><a href="preprocess.html#preprocessing-pipline"><i class="fa fa-check"></i><b>6.1.3</b> Preprocessing pipline</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="preprocess.html"><a href="preprocess.html#preprocess_mutations"><i class="fa fa-check"></i><b>6.2</b> Preprocessing for somatic mutation analysis</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="preprocess.html"><a href="preprocess.html#pileup-with-cellsnp-lite"><i class="fa fa-check"></i><b>6.2.1</b> Pileup with cellsnp-lite</a></li>
<li class="chapter" data-level="6.2.2" data-path="preprocess.html"><a href="preprocess.html#clonal-analysis-with-mquad"><i class="fa fa-check"></i><b>6.2.2</b> Clonal analysis with MQuad</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="preprocess.html"><a href="preprocess.html#wsl_install"><i class="fa fa-check"></i><b>6.3</b> (Optional) Install Windows Subsystem for Linux</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="preprocess.html"><a href="preprocess.html#what-is-the-windows-subsystem-for-linux-wsl"><i class="fa fa-check"></i><b>6.3.1</b> What is the Windows Subsystem for Linux (WSL)?</a></li>
<li class="chapter" data-level="6.3.2" data-path="preprocess.html"><a href="preprocess.html#manual-installation-steps"><i class="fa fa-check"></i><b>6.3.2</b> Manual Installation Steps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">HKU Single-cell Workshop (Modules 4.2 &amp; 5)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="references" class="section level1 unnumbered">
<h1>References</h1>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="preprocess.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["SingleCell Workshop.pdf", "SingleCell Workshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
