<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Trajectory inference and RNA velocity | HKU Single-cell Workshop (Modules 4.2 &amp; 5)</title>
  <meta name="description" content="Cell trajectory inference; Cellular genetic analysis" />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Trajectory inference and RNA velocity | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Trajectory inference and RNA velocity | HKU Single-cell Workshop (Modules 4.2 &amp; 5)" />
  
  <meta name="twitter:description" content="Cell trajectory inference; Cellular genetic analysis" />
  

<meta name="author" content="Yuanhua Lab: Mingze Gao, Rongting Huang, Xianjie Huang, Aaron Kwok, Zhuoxuan Li, Chen Qiao, Yuanhua Huang" />


<meta name="date" content="2021-07-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="env-pre.html"/>
<link rel="next" href="somatic-mutation-analysis-in-single-cells.html"/>
<script src="libs/header-attrs-2.9/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">SingleCell Workshop 2021</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#contents-of-this-part"><i class="fa fa-check"></i><b>1.1</b> Contents of this part</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#other-modules-of-this-workshop"><i class="fa fa-check"></i><b>1.2</b> Other modules of this workshop</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="env-pre.html"><a href="env-pre.html"><i class="fa fa-check"></i><b>2</b> Prerequisites</a>
<ul>
<li class="chapter" data-level="2.1" data-path="env-pre.html"><a href="env-pre.html#r-packages"><i class="fa fa-check"></i><b>2.1</b> R packages</a></li>
<li class="chapter" data-level="2.2" data-path="env-pre.html"><a href="env-pre.html#python-package-scvelo-with-anaconda"><i class="fa fa-check"></i><b>2.2</b> Python package scvelo with Anaconda</a>
<ul>
<li class="chapter" data-level="2.2.1" data-path="env-pre.html"><a href="env-pre.html#conda_install_windows"><i class="fa fa-check"></i><b>2.2.1</b> Installation on Windows</a></li>
<li class="chapter" data-level="2.2.2" data-path="env-pre.html"><a href="env-pre.html#conda_install_linux"><i class="fa fa-check"></i><b>2.2.2</b> Installation on Linux</a></li>
<li class="chapter" data-level="2.2.3" data-path="env-pre.html"><a href="env-pre.html#conda_install_mac"><i class="fa fa-check"></i><b>2.2.3</b> Installation on macOS</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html"><i class="fa fa-check"></i><b>3</b> Trajectory inference and RNA velocity</a>
<ul>
<li class="chapter" data-level="3.1" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#introduction-of-trajectory-inference"><i class="fa fa-check"></i><b>3.1</b> Introduction of trajectory inference</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#diffusion-map"><i class="fa fa-check"></i><b>3.1.1</b> Diffusion map</a></li>
<li class="chapter" data-level="3.1.2" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#monocle"><i class="fa fa-check"></i><b>3.1.2</b> Monocle</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#rna-velocity"><i class="fa fa-check"></i><b>3.2</b> RNA velocity</a></li>
<li class="chapter" data-level="3.3" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#rna-kinetics"><i class="fa fa-check"></i><b>3.3</b> RNA kinetics</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#unspliced-rna-indicates-transcriptional-speed"><i class="fa fa-check"></i><b>3.3.1</b> Unspliced RNA indicates transcriptional speed</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#use-scvelo-package-within-r"><i class="fa fa-check"></i><b>3.4</b> Use scVelo package within R</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#environment-and-preprocessing"><i class="fa fa-check"></i><b>3.4.1</b> Environment and preprocessing</a></li>
<li class="chapter" data-level="3.4.2" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#getting-started"><i class="fa fa-check"></i><b>3.4.2</b> Getting started</a></li>
<li class="chapter" data-level="3.4.3" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#data-preprocessing"><i class="fa fa-check"></i><b>3.4.3</b> Data Preprocessing</a></li>
<li class="chapter" data-level="3.4.4" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#pseudotime-based-on-diffusion-map"><i class="fa fa-check"></i><b>3.4.4</b> Pseudotime based on Diffusion Map</a></li>
<li class="chapter" data-level="3.4.5" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#compute-velocity-and-velocity-graph"><i class="fa fa-check"></i><b>3.4.5</b> Compute velocity and velocity graph</a></li>
<li class="chapter" data-level="3.4.6" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#diffusion-map-pseudotime-with-velocity"><i class="fa fa-check"></i><b>3.4.6</b> Diffusion-map Pseudotime with velocity</a></li>
<li class="chapter" data-level="3.4.7" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#plot-results"><i class="fa fa-check"></i><b>3.4.7</b> Plot Results</a></li>
<li class="chapter" data-level="3.4.8" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#interprete-velocity"><i class="fa fa-check"></i><b>3.4.8</b> Interprete Velocity</a></li>
<li class="chapter" data-level="3.4.9" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#velocity-in-cycling-progenitors"><i class="fa fa-check"></i><b>3.4.9</b> Velocity in cycling progenitors</a></li>
<li class="chapter" data-level="3.4.10" data-path="trajectory-inference-and-rna-velocity.html"><a href="trajectory-inference-and-rna-velocity.html#dynamical-mode-and-related-analysis"><i class="fa fa-check"></i><b>3.4.10</b> Dynamical Mode and related analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html"><i class="fa fa-check"></i><b>4</b> Somatic mutation analysis in single cells</a>
<ul>
<li class="chapter" data-level="4.1" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#introduction-1"><i class="fa fa-check"></i><b>4.1</b> Introduction</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#choice-of-protocols"><i class="fa fa-check"></i><b>4.1.1</b> Choice of protocols</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#snv-analysis"><i class="fa fa-check"></i><b>4.2</b> SNV analysis</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#call-somatic-variants-and-genotype-single-cells"><i class="fa fa-check"></i><b>4.2.1</b> Call somatic variants and genotype single cells</a></li>
<li class="chapter" data-level="4.2.2" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#visualize-variat-allele-fequency"><i class="fa fa-check"></i><b>4.2.2</b> Visualize variat allele fequency</a></li>
<li class="chapter" data-level="4.2.3" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#use-extra-information-on-clonal-tree"><i class="fa fa-check"></i><b>4.2.3</b> Use extra information on clonal tree</a></li>
<li class="chapter" data-level="4.2.4" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#run-cardelino-for-inferring-clonal-structure"><i class="fa fa-check"></i><b>4.2.4</b> Run cardelino for inferring clonal structure</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#mtsnv-analysis"><i class="fa fa-check"></i><b>4.3</b> mtSNV analysis</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#visualise-allele-frequency"><i class="fa fa-check"></i><b>4.3.1</b> Visualise allele frequency</a></li>
<li class="chapter" data-level="4.3.2" data-path="somatic-mutation-analysis-in-single-cells.html"><a href="somatic-mutation-analysis-in-single-cells.html#infer-clonal-structure-with-mtdna-variants"><i class="fa fa-check"></i><b>4.3.2</b> Infer clonal structure with mtDNA variants</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html"><i class="fa fa-check"></i><b>5</b> Copy number variation estimation from scRNA-seq</a>
<ul>
<li class="chapter" data-level="5.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#method-infercnv"><i class="fa fa-check"></i><b>5.1</b> Method: inferCNV</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#install-infercnv"><i class="fa fa-check"></i><b>5.1.1</b> install inferCNV</a></li>
<li class="chapter" data-level="5.1.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#getting-started-1"><i class="fa fa-check"></i><b>5.1.2</b> getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#application-on-tnbc1"><i class="fa fa-check"></i><b>5.2</b> Application on TNBC1</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#data-description"><i class="fa fa-check"></i><b>5.2.1</b> data description</a></li>
<li class="chapter" data-level="5.2.2" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#run-infercnv"><i class="fa fa-check"></i><b>5.2.2</b> run inferCNV</a></li>
<li class="chapter" data-level="5.2.3" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#infercnv-result"><i class="fa fa-check"></i><b>5.2.3</b> inferCNV result</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="copy-number-variation-estimation-from-scrna-seq.html"><a href="copy-number-variation-estimation-from-scrna-seq.html#ref"><i class="fa fa-check"></i><b>5.3</b> ref</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html"><i class="fa fa-check"></i><b>6</b> Preprocessing of dataset</a>
<ul>
<li class="chapter" data-level="6.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#preprocessing-for-rna-velocity"><i class="fa fa-check"></i><b>6.1</b> Preprocessing for RNA Velocity</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#list-of-packages-aligned-in-pipeline"><i class="fa fa-check"></i><b>6.1.1</b> List of packages aligned in pipeline:</a></li>
<li class="chapter" data-level="6.1.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#installation-of-packagessoftwares"><i class="fa fa-check"></i><b>6.1.2</b> Installation of packages/softwares</a></li>
<li class="chapter" data-level="6.1.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#preprocessing-pipline"><i class="fa fa-check"></i><b>6.1.3</b> Preprocessing pipline</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#preprocessing-for-somatic-mutation-analysis"><i class="fa fa-check"></i><b>6.2</b> Preprocessing for somatic mutation analysis</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#pileup-with-cellsnp-lite"><i class="fa fa-check"></i><b>6.2.1</b> Pileup with cellsnp-lite</a></li>
<li class="chapter" data-level="6.2.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#clonal-analysis-with-mquad"><i class="fa fa-check"></i><b>6.2.2</b> Clonal analysis with MQuad</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#wsl_install"><i class="fa fa-check"></i><b>6.3</b> (Optional) Install Windows Subsystem for Linux</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#what-is-the-windows-subsystem-for-linux-wsl"><i class="fa fa-check"></i><b>6.3.1</b> What is the Windows Subsystem for Linux (WSL)?</a></li>
<li class="chapter" data-level="6.3.2" data-path="preprocessing-of-dataset.html"><a href="preprocessing-of-dataset.html#manual-installation-steps"><i class="fa fa-check"></i><b>6.3.2</b> Manual Installation Steps</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">HKU Single-cell Workshop (Modules 4.2 &amp; 5)</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="trajectory-inference-and-rna-velocity" class="section level1" number="3">
<h1><span class="header-section-number">Chapter 3</span> Trajectory inference and RNA velocity</h1>
<div id="introduction-of-trajectory-inference" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Introduction of trajectory inference</h2>
<p>Generally, an ideal dynamical analysis of a biological process is performing
time-series experiment. However, the bottleneck may exist in the time resolution
of the experiment, for example it is difficult to have a resolution shorter than
a minute for RNA metabolic experiments. On the other hand, a snapshot of a
population of cells (namely only single time point) can still capture different
states of a biological process given cells have different speed in the process
or even different fates. Usually, a snapshot representing a relatively transient
dynamical process.</p>
<p>Trajectory inference is type of computational methods to infer the latent time
(or ordering) of a group of cells in a certain biological processes, for example
cell cycle, cell differentiation, or response to an external stimuli.
According to <a href="https://en.wikipedia.org/wiki/Trajectory_inference">Wikipedia</a>,</p>
<blockquote>
<p><em>“Trajectory inference or pseudotemporal ordering is a computational technique used in single-cell transcriptomics to determine the pattern of a dynamic process experienced by cells and then arrange cells based on their progression through the process.”</em></p>
</blockquote>
<p>Trajectory inference remains one of central interests in single-cell
transcriptomics studies. Its popularity can be seen from large number of
recently developed tools: in a recent benchmarking study published in Nature
Biotechnology
<a href="https://www.nature.com/articles/s41587-019-0071-9">Saelens et al 2019</a>, 45 out
of more than 70 tools have been evaluated.</p>
<p>The topology of the trajectory can be linear, multifurcation, cycle or even
more complex. The performance of each method on different tasks or data sets
varies a lot. We leave the readers to the original benchmark paper metioned
above.</p>
<div class="figure">
<img src="images/trajectory-fig1.png" alt="" />
<p class="caption">Figure 3.1. Different trajectory topologies and its comparison. Fig. adapted from Saelens et al, 2019</p>
</div>
<p>Here, we only introduce two commonly used methods. Note, all above trajecotry
inference methods are based on the current state of transcription and the
distance between cell are symmetric, namely distance from cell A to cell B is
the same as from cell B to cell A. Therefore, the direction of the inferred
trajectory is either lack (namely require users to specify) or less reliable if
using default setting.</p>
<div id="diffusion-map" class="section level3" number="3.1.1">
<h3><span class="header-section-number">3.1.1</span> Diffusion map</h3>
<p><a href="https://en.wikipedia.org/wiki/Diffusion_map">Diffusion map</a> is a type of
non-linear dimension reduction methods, and is recently applied to estimate the
pseudotime in single-cell transcriptomic data (<a href="https://www.nature.com/articles/nmeth.3971">Haghverdi et al 2016</a>).</p>
<p>As shown in Fig. 3.1 (adapted from Haghverdi et al 2016), this algorithm works
with three steps:</p>
<ol style="list-style-type: decimal">
<li>computing the overlap of local kernels at the expression levels of cells x
and y.</li>
<li>Diffusion pseudotime dpt(x,y) approximates the geodesic distance
between x and y on the mapped manifold,</li>
<li>Branching points are identified as points where anticorrelated distances from
branch ends become correlated.</li>
</ol>
<div class="figure">
<img src="images/trajectory-fig2.png" alt="" />
<p class="caption">Figure 3.1. Diffusion-map pseudotime. Fig. adapted from Haghverdi et al 2016</p>
</div>
</div>
<div id="monocle" class="section level3" number="3.1.2">
<h3><span class="header-section-number">3.1.2</span> Monocle</h3>
<p>Another widely used tool is
<a href="https://cole-trapnell-lab.github.io/monocle3/">Monocle (now v3)</a>. One of cool
features that Monocle3 offers is the branching in the trajectory. On the other
hand, it generally requires users to specify the start (or end) point in the
trajectory.</p>
<div class="figure">
<img src="images/trajectory-fig3.png" style="width:60.0%" alt="" />
<p class="caption">Figure 3.1. Example results for Monocle3. Fig. adapted from its tutorial</p>
</div>
</div>
</div>
<div id="rna-velocity" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> RNA velocity</h2>
</div>
<div id="rna-kinetics" class="section level2" number="3.3">
<h2><span class="header-section-number">3.3</span> RNA kinetics</h2>
<p>As briefly mentioned that the trajectory inferred from the transcriptome ofter
suffer from lack of automatically detected directionality of the trajectory.
Very recently, the RNA velocity is a newly introduced concept to use the
unspliced RNAs to indicate the transcriptional kinetic activity
(<a href="https://www.nature.com/articles/s41586-018-0414-6">La Manno et al, 2018</a>), and
a recent extension to full dynamical model
(<a href="https://www.nature.com/articles/s41587-020-0591-3">Bergen et al, 2020</a>).</p>
<p>Let us first review the RNA metabolic process and its intrinsic dynamics:
- RNA is synthesized at a rate of alpha
- RNA is spliced to remove introns and reformed into mature mRNA;
splicing rate beta
- Mature mRNA is degraded after functioning at a rate of gamma</p>
<p>In steady state, the unspliced and spliced is at an equilibrium, namely the
newly synthesized is exactly equal to the spliced and also the degradated.
Therefore, the break of the equilibrium state between unspliced and spliced RNAs
is highly informative to indicate whether a gene is in the induction or
suppression state.</p>
<div class="figure">
<img src="images/trajectory-fig4.png" alt="" />
<p class="caption">Figure 3.4. RNA kinetics and RNA velocity.Fig. adapted from Haghverdi et al 2016</p>
</div>
<div id="unspliced-rna-indicates-transcriptional-speed" class="section level3" number="3.3.1">
<h3><span class="header-section-number">3.3.1</span> Unspliced RNA indicates transcriptional speed</h3>
<p>Usually, the proportion of unspliced RNA is very low in RNA-seq, particularly
because most protocols is based on Poly-A enriched RNAs. Namely, in principle,
only RNAs reaching 3’ of the gene body can be captured. However, the unspliced
RNAs still can be observed at a substantial proportion, usually covering 15-25%.</p>
<p>Recently, there are also a few technologies have been introduced to enrich the
nascent RNAs in single-cell protocols, e.g.,
<a href="https://www.nature.com/articles/s41586-019-1369-y">scSLAM-seq</a> and
<a href="https://www.nature.com/articles/s41592-020-0935-4">scNT-seq</a>.</p>
<p>Though some researchers remain skeptical on the use of RNA velocity, possibly
because the high noise-by-signal ratio, the unprecedent capability to
automatically detect the trajectory direction attracts much attention. On the
other hands, more accurate and robust methods are under rapid development, to
make this analysis tool more widely applicable.</p>
</div>
</div>
<div id="use-scvelo-package-within-r" class="section level2" number="3.4">
<h2><span class="header-section-number">3.4</span> Use scVelo package within R</h2>
<p>There are two major tools for analysing RNA velocity:</p>
<ul>
<li><a href="http://velocyto.org">velocyto</a> is based on a steady-state model for
estimating RNA velocity. It supports both Python and R.</li>
<li><a href="https://scvelo.readthedocs.io">scVelo</a> introced an extended dynamical model
for estimating RNA velocity, and various of utility functions. However, it
only supports Python.</li>
</ul>
<p>Here, we will introduce <code>reticulate</code> to use Python package in R (sounds cool,
right, and you will find it handy!)</p>
<div id="environment-and-preprocessing" class="section level3" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Environment and preprocessing</h3>
<p>In order or use <code>scVelo</code>, we need to install it. Please follow Chapter 2.1.2 to
install Python and <code>scVelo==0.2.2</code>. As mentioned there, we recommend using
<a href="https://www.anaconda.com/products/individual">Anaconda</a> to create Python and
conda environment.</p>
<p>Additionally, the extration of unspliced RNA is a bit tricky, and not widely
supported. There are three options we have tried (there maybe more, e.g.,
kallisto bustools):</p>
<ul>
<li><a href="http://velocyto.org">velocyto.py</a>: the earliest software for this purpose.
Generally not computationally efficient, possible due to written in Python.
For unknown reason, the proportion of unspliced RNA is unrealisticly high for
5’ 10x Genomics data.</li>
<li><a href="https://dropest.readthedocs.io">dropEst</a>: as implemented in C/C++, it is much
more efficient. It also returns more reasonable proportion of unspliced RNAs
for 5’ 10x Genomics data</li>
<li><a href="https://github.com/alexdobin/STAR/blob/master/docs/STARsolo.md">STAR-solo</a>:
new extension for the popular STAR. Benefits: efficient and one step for reads
alignment and counting of unspliced RNA (<strong>Recommended option</strong>)</li>
</ul>
</div>
<div id="getting-started" class="section level3" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Getting started</h3>
<p>First, let’s use load <code>reticulate</code>:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="trajectory-inference-and-rna-velocity.html#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># install.packages(&quot;reticulate&quot;)</span></span>
<span id="cb7-2"><a href="trajectory-inference-and-rna-velocity.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reticulate)</span></code></pre></div>
<p>Then use reticulate to import Python packages <code>scvelo</code> and <code>matplotlib</code>:</p>
<pre echo="TRUE," message="TRUE"><code>use_condaenv(&quot;sgcell&quot;)
plt &lt;- import(&quot;matplotlib.pyplot&quot;, as = &quot;plt&quot;)
scv &lt;- import (&quot;scvelo&quot;)

scv$logging$print_version()
scv$settings$presenter_view = TRUE
scv$settings$verbosity = 3
scv$settings$set_figure_params(&quot;scvelo&quot;) </code></pre>
<p>In this tutorial, we will download and use the built-in pancreas dataset to
demonstrate the useage of RNA velocity analysis. The dataset will be
automatically downloaded to <code>./data/Pancreas/endocrinogenesis_day15.h5ad</code>
(52.5MB).</p>
<p>Endocrine development in pancreas lineage has four major fates: <span class="math inline">\(\alpha\)</span>, <span class="math inline">\(\beta\)</span>, <span class="math inline">\(\delta\)</span>, <span class="math inline">\(\epsilon\)</span>. Dataset is obtained and subseted from
<a href="https://journals.biologists.com/dev/article/146/12/dev173849/19483/Comprehensive-single-cell-mRNA-profiling-reveals-a">Bastidas-Ponce et al. (2018)</a>
by the scVelo team. In this</p>
<pre><code>adata &lt;- scv$datasets$pancreas()
adata</code></pre>
<p>Note, to run velocity analysis on your own data, read your file with
<code>adata = scv$read(file_path)</code>.
If you want to save model and parameters after processing, run the following command, <code>adata$write(file_path, compression = 'gzip')</code>.</p>
<p>Now, let’s have a look of the proportions of unspliced/spliced mRNA reads,
UMAP embedding and cluster annotations can be printed and visualized using
built-in functions.</p>
<pre><code>scv$utils$show_proportions(adata)
scv$pl$proportions(adata, figsize=c(10, 3), show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig1.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig1.png" style="width:80.0%" alt="" />
<p class="caption">RNA-velo-fig2</p>
</div>
<pre><code>scv$pl$scatter(adata, legend_loc = &quot;best&quot;, size = 50, title = &quot;Pancreas Celltype&quot;, show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig2.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig2.png" style="width:75.0%" alt="" />
<p class="caption">RNA-velo-fig2</p>
</div>
</div>
<div id="data-preprocessing" class="section level3" number="3.4.3">
<h3><span class="header-section-number">3.4.3</span> Data Preprocessing</h3>
<p>Preprocessing contains:<br />
- gene selection by detection (detected with a minimum number of counts)<br />
- high variability (dispersion)<br />
- normalizing every cell by its initial size and logarithmizing X</p>
<p>First and second order moments are also computed (mean, uncentered variance for deterministic, stochastic mode respectively) among nearest neighbors in PCA space.</p>
<pre><code>scv$pp$filter_and_normalize(adata, min_shared_counts = as.integer(20), n_top_genes = as.integer(2000))
scv$pp$moments(adata, n_pcs = as.integer(30), n_neighbors = as.integer(30))</code></pre>
</div>
<div id="pseudotime-based-on-diffusion-map" class="section level3" number="3.4.4">
<h3><span class="header-section-number">3.4.4</span> Pseudotime based on Diffusion Map</h3>
<p>Pseudotime, a part of standardized scRNA-seq analysis pipeline, is also implemented in this package, and can be compared with the <code>colorize("latent time", "#316A9E")</code> introduced in dynamical mode.</p>
<pre><code>adata$uns$data$iroot &lt;- which.min(adata$obsm[&#39;X_umap&#39;][, 1])
scv$tl$diffmap(adata)
scv$tl$dpt(adata)

scv$pl$scatter(adata, color = &#39;dpt_pseudotime&#39;, title = &#39;pseudotime&#39;, color_map = &#39;gnuplot&#39;, colorbar = TRUE, rescale_color = c(0,1), perc=c(2, 98), show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig3.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig3.png" style="width:75.0%" alt="" />
<p class="caption">RNA-velo-fig3</p>
</div>
</div>
<div id="compute-velocity-and-velocity-graph" class="section level3" number="3.4.5">
<h3><span class="header-section-number">3.4.5</span> Compute velocity and velocity graph</h3>
<p>scVelo has incorporated 3 modes for velocity estimation:<br />
- Deterministic<br />
- Stochastic<br />
- Dynamical</p>
<p>For deterministic and stochastic mode, the gene-specific velocities are obtained by fitting linear regression ratio (constant transcriptional state) between unspliced/spliced mRNA abundances.</p>
<p>Under linear assumptions, how the <code>colorize("observed abundances deviate from the steady state", "#b22e5b")</code> regression line is velocity.</p>
<pre><code>scv$tl$velocity(adata, mode = &quot;stochastic&quot;)</code></pre>
<p>To calculate velocity graph, we need to run <code>colorize("velocity_graph()", "#316A9E")</code>. Velocity graph is the cosine correlation of potential cell transitions with velocity vector in high dimensional space. It summarizes the possible cell transition states and has dimension of <span class="math inline">\({n}_{obs} * {n}_{obs}\)</span>.</p>
<pre><code>scv$tl$velocity_graph(adata, sqrt_transform = TRUE)</code></pre>
</div>
<div id="diffusion-map-pseudotime-with-velocity" class="section level3" number="3.4.6">
<h3><span class="header-section-number">3.4.6</span> Diffusion-map Pseudotime with velocity</h3>
<p>Pseudotime, a part of standardized scRNA-seq analysis pipeline, is also implemented in this package, and can be compared with the <code>colorize("latent time", "#316A9E")</code> introduced in dynamical mode.</p>
<pre><code>scv$tl$velocity_pseudotime(adata)
scv$pl$scatter(adata, color = &#39;velocity_pseudotime&#39;, cmap = &#39;gnuplot&#39;, show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig4.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig4.png" style="width:75.0%" alt="" />
<p class="caption">RNA-velo-fig4</p>
</div>
<pre><code>scv$pl$scatter(adata, x = &quot;velocity_pseudotime&quot;, y = c(&#39;Actn4&#39;, &#39;Ppp3ca&#39;, &#39;Cpe&#39;, &#39;Nnat&#39;), fontsize = 10, size = 10, legend_loc = &#39;best&#39;, color = &#39;clusters&#39;, figsize=c(12, 10), show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig5.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig5.png" alt="" />
<p class="caption">RNA-velo-fig5</p>
</div>
</div>
<div id="plot-results" class="section level3" number="3.4.7">
<h3><span class="header-section-number">3.4.7</span> Plot Results</h3>
<p>Velocities are projected onto the specified embedding <code>colorize("basis", "#316A9E")</code> and can be visualized in one of the three ways:<br />
- On single cell level<br />
- On grid level<br />
- Streamlines, which is most commonly used</p>
<pre><code>scv$pl$velocity_embedding_stream(adata, basis = &quot;umap&quot;, color = &quot;clusters&quot;, legend_loc = &quot;best&quot;, dpi = 150, show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig6.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig6.png" style="width:75.0%" alt="" />
<p class="caption">RNA-velo-fig5</p>
</div>
<pre><code>scv$pl$velocity_embedding(adata, basis = &quot;umap&quot;, arrow_length = 3, arrow_size = 2, dpi = 150, show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig7.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig7.png" style="width:75.0%" alt="" />
<p class="caption">RNA-velo-fig7</p>
</div>
</div>
<div id="interprete-velocity" class="section level3" number="3.4.8">
<h3><span class="header-section-number">3.4.8</span> Interprete Velocity</h3>
<p>We could also examine the phase portraits of interested genes to understand how inferred directions are supported by particular genes.</p>
<pre><code>scv$pl$velocity(adata, c(&quot;Cpe&quot;, &quot;Gnao1&quot;, &quot;Ins2&quot;, &quot;Adk&quot;), ncols = 2, show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig8.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig8.png" alt="" />
<p class="caption">RNA-velo-fig8</p>
</div>
<p>Positive velocity indicates that a gene is up-regulated, which occurs for cells that show higher abundance of unspliced mRNA for that gene than expected in steady state. Conversely, negative velocity indicates that a gene is down-regulated.</p>
</div>
<div id="velocity-in-cycling-progenitors" class="section level3" number="3.4.9">
<h3><span class="header-section-number">3.4.9</span> Velocity in cycling progenitors</h3>
<p>The cell cycle detected by RNA velocity, is biologically affirmed by cell cycle scores (standardized scores of mean expression levels of phase marker genes).</p>
<pre><code>scv$tl$score_genes_cell_cycle(adata)
scv$pl$scatter(adata, color_gradients = c(&quot;S_score&quot;, &quot;G2M_score&quot;), smooth = TRUE, perc = c(5, 95), legend_loc=&quot;upper center&quot;,  show=FALSE)
plt$savefig(&#39;images/RNA-velo-fig9.png&#39;)</code></pre>
<div class="figure">
<img src="images/RNA-velo-fig9.png" alt="" />
<p class="caption">RNA-velo-fig9</p>
</div>
</div>
<div id="dynamical-mode-and-related-analysis" class="section level3" number="3.4.10">
<h3><span class="header-section-number">3.4.10</span> Dynamical Mode and related analysis</h3>
<p>Dynamical mode does not necessarily rely on linear assumptions, instead it consider gene-specific rates of transcription, splicing and degradation rates as well as transient cell-states.</p>
<p>To use dynamical mode, we just need to call <code>colorize("revocer_dynamics()", "#316A9E")</code> before computing velocity.</p>
<pre><code>scv$tl$recover_dynamics(adata)

scv$tl$velocity(adata, mode = &quot;dynamical&quot;)
scv$tl$velocity_graph(adata)</code></pre>
<p>Latent time of the underlying cellular processes can be recovered via dynamical mode. Based on cells’ transcriptional dynamics, latent time approximates the real time experienced by cells as they differentiate.</p>
<p>Latent time could distinguish temporal position more compared with pseudotime.</p>
<pre><code>scv$tl$latent_time(adata)

scv$pl$scatter(adata, color = &quot;latent_time&quot;, color_map = &quot;gnuplot&quot;, size = 80, basis = &quot;umap&quot;)
scv$pl$scatter(adata, x = &quot;latent_time&quot;, y = c(&#39;Actn4&#39;, &#39;Ppp3ca&#39;, &#39;Cpe&#39;, &#39;Nnat&#39;), fontsize = 10, size = 10, legend_loc = &#39;best&#39;, color = &#39;clusters&#39;)</code></pre>
<p>We could also pull out the top genes driving the dynamic RNA velocities.</p>
<pre><code>topgenes &lt;- adata$var[&quot;fit_likelihood&quot;]
topgenes_vals &lt;- topgenes[, 1]
names(topgenes_vals) &lt;- rownames(topgenes)
topgenes_vals &lt;- sort(topgenes_vals, decreasing = TRUE)
head(topgenes_vals)

scv$pl$scatter(adata, basis = names(topgenes_vals)[1:5], ncols = 5, frameon = FALSE)

scv$pl$heatmap(adata, var_names = names(topgenes_vals), tkey = &#39;latent_time&#39;, n_convolve = as.integer(100), col_color = &#39;clusters&#39;)</code></pre>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="env-pre.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="somatic-mutation-analysis-in-single-cells.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["SingleCell Workshop.pdf", "SingleCell Workshop.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
